<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<title>Genome Annotation: Hands-on Training: All in One View</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<script src="../assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="../assets/styles.css">
<script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../favicons/rcac/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicons/rcac/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="../favicons/rcac/favicon-16x16.png">
<link rel="manifest" href="../favicons/rcac/site.webmanifest">
<link rel="mask-icon" href="../favicons/rcac/safari-pinned-tab.svg" color="#5bbad5">
<meta name="msapplication-TileColor" content="#da532c">
<meta name="theme-color" media="(prefers-color-scheme: light)" content="white">
<meta name="theme-color" media="(prefers-color-scheme: dark)" content="black">
</head>
<body>
    <header id="top" class="navbar navbar-expand-md top-nav rcac"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="rcac-logo" alt="rcac" src="../assets/images/rcac-logo.svg">
</div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text">
<li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul>
</li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
<li><button class="dropdown-item" type="button" onclick="window.location.href='../aio.html';">Learner View</button></li>
        </ul>
</div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav rcac" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="rcac" src="../assets/images/rcac-logo-sm.svg">
</div>
    <div class="lesson-title-md">
      Genome Annotation: Hands-on Training
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
<li class="nav-item">
          <span class="lesson-title">
            Genome Annotation: Hands-on Training
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown">
<hr>
<li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul>
</li>
      </ul>
</div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="../instructor/aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div>
<!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Genome Annotation: Hands-on Training
</div>

<aside class="col-md-12 lesson-progress"><div style="width: %" class="percentage">
    %
  </div>
  <div class="progress rcac">
    <div class="progress-bar rcac" role="progressbar" style="width: %" aria-valuenow="" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text">
<li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul>
</li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../aio.html">Learner View</a>
                      </div>
                    </div>
                  </div>
<!--/div.accordion-item-->
                </div>
<!--/div.accordion-flush-->
              </div>
<!--div.sidenav-view-selector -->
            </div>
<!--/div.col -->

            <hr>
</div>
<!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="introduction.html">1. Introduction to Genome Annotation</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="annotation-strategies.html">2. Annotation Strategies</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="annotation-setup.html">3. Annotation Setup</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="gene-annotation-with-braker.html">4. Annotation using BRAKER</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="gene-annotation-with-helixer.html">5. Annotation using Helixer</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="gene-annotation-with-easel.html">6. Annotation using Easel</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="functional-annotation-with-entap.html">7. Functional annotation using EnTAP</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="annotation-assesment.html">8. Annotation Assesment</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width">
<div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul>
<li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <hr>
<li><a class="dropdown-item" href="reference.html">Reference</a></li>
                    </ul>
</div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources">
<a href="../instructor/aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none">
<div class="d-grid gap-1">

            </div>
          </div>
<!-- /div.accordion -->
        </div>
<!-- /div.sidebar-inner -->
      </nav>
</div>
<!-- /div.sidebar -->
  </div>
<!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-extra.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <main id="main-content" class="main-content"><div class="container lesson-content">
        
        
<section id="aio-introduction"><p>Content from <a href="introduction.html">Introduction to Genome Annotation</a></p>
<hr>
<p>Last updated on 2025-02-22 |

        <a href="https://rcac-bioinformatics.github.io/genome-annotation/edit/main/episodes/introduction.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 20 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What is genome annotation, and why is it important?<br>
</li>
<li>What are the different types of genome annotation?<br>
</li>
<li>What challenges make genome annotation difficult?<br>
</li>
<li>What data sources are used to improve annotation accuracy?<br>
</li>
<li>How does the annotation process fit into genomics research?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Define genome annotation and its role in genomics research<br>
</li>
<li>Differentiate between structural and functional annotation<br>
</li>
<li>Identify key challenges in genome annotation and how they impact
accuracy<br>
</li>
<li>Understand the types of data used in annotation and their
significance<br>
</li>
<li>Provide an overview of the annotation workflow used in this
workshop</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="what-is-genome-annotation">What Is Genome Annotation?<a class="anchor" aria-label="anchor" href="#what-is-genome-annotation"></a>
</h2>
<hr class="half-width">
<ul>
<li>Genome annotation is the process of identifying and labeling
functional elements in a genome, such as genes, regulatory regions, and
non-coding RNAs.<br>
</li>
<li>It involves predicting gene structures, including exons, introns,
and untranslated regions, using computational and experimental
data.<br>
</li>
<li>Functional annotation assigns biological roles to predicted genes
based on sequence similarity, protein domains, and pathway
associations.<br>
</li>
<li>Accurate annotation is essential for understanding genome function,
evolutionary relationships, and applications in research and
biotechnology.</li>
</ul>
<figure><img src="https://github.com/user-attachments/assets/fd2746d7-077a-404d-b690-a4ac7bcb45a8" alt="The structure of a eukaryotic protein-coding gene" class="figure mx-auto d-block"><div class="figcaption">The structure of a eukaryotic protein-coding
gene</div>
</figure><p><a href="https://en.wikipedia.org/wiki/Gene_structure" class="external-link">source</a></p>
<div id="why-is-annotation-important" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="why-is-annotation-important" class="callout-inner">
<h3 class="callout-title">Why Is Annotation Important?</h3>
<div class="callout-content">
<ul>
<li>Annotation provides a functional map of the genome, helping
researchers identify genes and regulatory elements.<br>
</li>
<li>It enables comparative genomics, allowing insights into evolution,
gene conservation, and species-specific adaptations.<br>
</li>
<li>Accurate gene annotation is essential for biomedical and
agricultural research, including disease studies and crop
improvement.<br>
</li>
<li>Functional annotations link genes to biological processes, aiding in
pathway analysis and experimental validation.</li>
</ul>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="types-of-genome-annotation">Types of Genome Annotation<a class="anchor" aria-label="anchor" href="#types-of-genome-annotation"></a>
</h2>
<hr class="half-width">
<ul>
<li>
<strong>Structural annotation</strong> identifies genes, exon-intron
boundaries, untranslated regions (UTRs), and regulatory elements such as
promoters and enhancers.<br>
</li>
<li>
<strong>Functional annotation</strong> assigns biological roles to
genes based on sequence similarity, protein domains, gene ontology (GO),
and pathway associations.<br>
</li>
<li>
<strong>Repeat annotation</strong> detects transposable
elements/centromere repeats/knob sequences and other repetitive
sequences that prevent misannotation of genes.<br>
</li>
<li>
<strong>Epigenomic annotation</strong> integrates chromatin
accessibility, DNA methylation, and histone modifications to understand
gene regulation.</li>
<li>
<strong>Non-coding RNA annotation</strong> includes small RNAs
(miRNAs, siRNAs, piRNAs), long non-coding RNAs (lncRNAs), ribosomal RNAs
(rRNAs), and transfer RNAs (tRNAs).</li>
</ul></section><section><h2 class="section-heading" id="key-challenges-in-genome-annotation">Key Challenges in Genome Annotation<a class="anchor" aria-label="anchor" href="#key-challenges-in-genome-annotation"></a>
</h2>
<hr class="half-width">
<ul>
<li>
<strong>Incomplete or fragmented assemblies</strong>: Gaps and
misassemblies can lead to missing or incorrect gene predictions.<br>
</li>
<li>
<strong>Alternative splicing and isoforms</strong>: Complex gene
structures make it difficult to define a single gene model.<br>
</li>
<li>
<strong>Pseudogenes and transposable elements</strong>:
Distinguishing functional genes from non-functional sequences is
challenging.<br>
</li>
<li>
<strong>Low-quality or missing evidence</strong>: Incomplete
RNA-seq, proteomics, or evolutionary data can reduce annotation
accuracy.<br>
</li>
<li>
<strong>Annotation discrepancies across tools</strong>: Different
prediction methods may produce conflicting gene models.<br>
</li>
<li>
<strong>Evolutionary divergence</strong>: Gene structures and
functional elements can vary widely across species, complicating
homology-based annotation.</li>
</ul></section><section><h2 class="section-heading" id="data-sources-for-annotation">Data Sources for Annotation<a class="anchor" aria-label="anchor" href="#data-sources-for-annotation"></a>
</h2>
<hr class="half-width">
<p>Accurate genome annotation relies on multiple data sources to refine
gene predictions and validate functional elements. Different types of
experimental and computational evidence contribute to identifying gene
structures, transcript isoforms, coding potential, and functional
assignments.</p>
<p><strong>1. Long-read transcript sequencing (Iso-Seq)</strong><br>
- Captures full-length transcripts, helping define complete exon-intron
structures.<br>
- Useful for identifying alternative splicing events and novel
isoforms.</p>
<p><strong>2. Ribosome profiling (Ribo-Seq)</strong><br>
- Detects ribosome-protected mRNA fragments to identify actively
translated regions.<br>
- Helps differentiate coding from non-coding RNAs and refine start/stop
codons.</p>
<p><strong>3. Cap analysis gene expression (CAGE) and
RAMPAGE</strong><br>
- Maps transcription start sites by detecting 5’ capped mRNAs.<br>
- Provides precise information on promoter locations and alternative
start sites.</p>
<p><strong>4. Proteomics (Mass Spectrometry Data)</strong><br>
- Confirms protein-coding genes by detecting peptide fragments in
expressed proteins.<br>
- Helps validate translation and refine gene models.</p>
<p><strong>5. RNA sequencing (Short-read RNA-Seq)</strong><br>
- Provides gene expression levels and transcript coverage across
different conditions.<br>
- Essential for identifying intron-exon boundaries and alternative
splicing events.</p>
<p><strong>6. Comparative genomics (Synteny &amp;
Homology)</strong><br>
- Uses known gene models from related species to guide annotation.<br>
- Helps predict conserved genes and validate gene structures.</p>
<p><strong>7. Additional Data Sources</strong><br>
- <strong>Poly(A) site sequencing (PAS-Seq)</strong>: Defines transcript
end sites for more precise UTR annotation.<br>
- <strong>Chromatin accessibility data (ATAC-Seq, DNase-Seq)</strong>:
Identifies regulatory regions like enhancers and promoters.<br>
- <strong>Epigenomic data (ChIP-Seq, DNA methylation)</strong>: Provides
evidence for active and repressed genes.</p>
<figure><img src="https://github.com/user-attachments/assets/656dc410-af25-461e-bac2-36ccd92c9b28" alt="Integrating diverse datasets improves the accuracy of gene predictions, ensuring that both structural and functional annotations reflect real biological features." class="figure mx-auto d-block"><div class="figcaption">Integrating diverse datasets improves the
accuracy of gene predictions, ensuring that both structural and
functional annotations reflect real biological features.</div>
</figure></section><section><h2 class="section-heading" id="scope-of-this-workshop">Scope of This Workshop<a class="anchor" aria-label="anchor" href="#scope-of-this-workshop"></a>
</h2>
<hr class="half-width">
<p>In this workshop, we will focus on protein-coding gene annotation in
eukaryotic genomes using computational tools and available datasets. We
will annotate genes in a Arabidopsis genome using a combination of ab
initio gene prediction, homology-based methods, and RNA-seq evidence
methods, can compare the results to a reference annotation. The workflow
will cover data preparation, gene prediction, functional annotation, and
quality assessment steps to provide a comprehensive overview of the
annotation process.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Genome annotation identifies functional elements in a genome,
including genes, regulatory regions, and non-coding RNAs.</li>
<li>Structural annotation predicts gene structures, while functional
annotation assigns biological roles to genes.</li>
<li>Challenges in annotation include incomplete assemblies, alternative
splicing, pseudogenes, and data quality issues.</li>
<li>Data sources like RNA-seq, proteomics, comparative genomics, and
epigenomics improve annotation accuracy.</li>
<li>This workshop focuses on eukaryotic gene annotation using
computational tools and diverse datasets.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-annotation-strategies"><p>Content from <a href="annotation-strategies.html">Annotation Strategies</a></p>
<hr>
<p>Last updated on 2025-02-22 |

        <a href="https://rcac-bioinformatics.github.io/genome-annotation/edit/main/episodes/annotation-strategies.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What are the different strategies used for genome annotation?<br>
</li>
<li>How do various methods predict genes, and what are their strengths
and limitations?<br>
</li>
<li>What role do evidence play in improving gene predictions?<br>
</li>
<li>How do deep learning and large models enhance genome annotation
accuracy?<br>
</li>
<li>What tools are used in this workshop, and how do they fit into
different strategies?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand the principles behind different genome annotation
approaches.<br>
</li>
<li>Learn how intrinsic, extrinsic, and hybrid methods predict gene
structures.<br>
</li>
<li>Explore the role of multi-evidence approaches, including
transcriptomics, proteomics, and functional annotation.<br>
</li>
<li>Introduce modern machine learning and large model-based methods for
gene annotation.<br>
</li>
<li>Familiarize with annotation tools used in this workshop and their
specific applications.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr class="half-width">
<ul>
<li>Genome annotation identifies functional elements in a genome,
including protein-coding genes, non-coding rnas, and regulatory
regions<br>
</li>
<li>Annotation strategies vary between evidence-based, ab initio, and
hybrid approaches, each with strengths and limitations<br>
</li>
<li>Selecting the right method depends on genome complexity, available
transcriptomic and proteomic data, and the need for accuracy versus
sensitivity</li>
</ul></section><section><h2 class="section-heading" id="annotation-methods">Annotation Methods<a class="anchor" aria-label="anchor" href="#annotation-methods"></a>
</h2>
<hr class="half-width">
<p><strong>Intrinsic (Ab Initio) Methods</strong></p>
<ul>
<li>Predict genes based on sequence features without external
evidence.<br>
</li>
<li>Use statistical models such as hidden Markov models (HMMs) to
recognize coding regions, exon-intron boundaries, and gene
structures.<br>
</li>
<li>Rely on nucleotide composition patterns like codon usage, GC
content, and splice site signals.<br>
</li>
<li>Perform well in well-studied genomes but struggle with novel genes
and complex gene structures.</li>
</ul>
<p><strong>Extrinsic (Evidence-Based) Methods</strong></p>
<ul>
<li>Utilize external data such as RNA-seq, ESTs, and protein homology to
guide gene predictions.<br>
</li>
<li>Align transcriptomic and proteomic data to the genome to refine
exon-intron structures.<br>
</li>
<li>Improve annotation accuracy by incorporating evolutionary
conservation and known gene models.<br>
</li>
<li>Require high-quality reference datasets, making them less effective
in newly sequenced organisms with little prior data.</li>
</ul>
<p><strong>Hybrid Methods</strong></p>
<ul>
<li>Combine ab initio predictions with evidence-based approaches to
improve accuracy and completeness.<br>
</li>
<li>Use gene predictions from statistical models and refine them with
transcriptomic and proteomic data.<br>
</li>
<li>Balance sensitivity (detecting novel genes) and specificity
(avoiding false positives).<br>
</li>
<li>Often used in large-scale genome annotation projects where both
approaches complement each other.</li>
</ul>
<p><strong>Multi-Evidence based methods</strong></p>
<ul>
<li>Incorporate diverse biological data sources to improve gene
prediction accuracy.<br>
</li>
<li>Use <strong>ribosome profiling (Ribo-seq)</strong> and
<strong>proteomics</strong> to confirm protein-coding potential.<br>
</li>
<li>Leverage <strong>poly(A) signal presence</strong> to distinguish
mature transcripts from non-coding RNA.<br>
</li>
<li>Integrate <strong>RNA-seq, long-read sequencing, and EST
data</strong> to refine exon-intron structures.<br>
</li>
<li>Provide direct experimental validation, reducing reliance on purely
computational models.</li>
</ul>
<p><strong>Large Model-Based Methods</strong></p>
<ul>
<li>Utilize deep learning and large language models (LMMs) trained on
vast genomic datasets.<br>
</li>
<li>Predict gene structures by recognizing sequence patterns learned
from diverse organisms.<br>
</li>
<li>Adapt to various genome complexities without requiring manually
curated training data.<br>
</li>
<li>Identify non-canonical gene structures, alternative splicing events,
and novel genes.<br>
</li>
<li>Example: <strong>Helixer</strong>, a GPU-based model, predicts genes
using high-dimensional sequence patterns, improving over traditional
HMM-based methods.</li>
</ul>
<div id="what-do-we-annotate-in-the-genome" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="what-do-we-annotate-in-the-genome" class="callout-inner">
<h3 class="callout-title">What do we annotate in the genome?</h3>
<div class="callout-content">
<ul>
<li>
<strong>Protein-coding genes</strong>: Genes that encode proteins,
identified based on open reading frames (ORFs) and coding sequence
features.<br>
</li>
<li>
<strong>Pseudogenes</strong>: Non-functional remnants of once-active
genes that have accumulated mutations.<br>
</li>
<li>
<strong>Non-coding RNAs (ncRNAs)</strong>: Functional RNAs that do
not code for proteins, including tRNAs, rRNAs, and lncRNAs.<br>
</li>
<li>
<strong>Upstream Open Reading Frames (uORFs)</strong>: Small ORFs in
the 5’ untranslated region (UTR) that can regulate translation.<br>
</li>
<li>
<strong>Alternative Splicing Variants</strong>: Different transcript
isoforms from the same gene, generated by exon-intron
rearrangements.<br>
</li>
<li>
<strong>Repeat Elements and Transposons</strong>: Mobile DNA
sequences that can impact gene function and genome structure.<br>
</li>
<li>
<strong>Regulatory Elements</strong>: Promoters, enhancers, and
other sequences controlling gene expression.</li>
</ul>
<p>For this workshop, we focus only on <strong>protein-coding
genes</strong> and their annotation.</p>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="annotation-workflows-used">Annotation Workflows Used<a class="anchor" aria-label="anchor" href="#annotation-workflows-used"></a>
</h2>
<hr class="half-width">
<p><strong>1. BRAKER3</strong>: Combines GeneMark-ET and AUGUSTUS to
predict genes in eukaryotic genomes.</p>
<ul>
<li>Integrates GeneMark and AUGUSTUS with RNA-Seq and/or protein
evidence.</li>
<li>Supports various data types or runs without them.</li>
<li>Fully automated training and gene prediction.</li>
<li>Requires evidence data for high accuracy; performs poorly without
it.</li>
<li>Struggles with scalability on larger genomes.</li>
</ul>
<div class="tabs">
<nav><div id="nav-tab-1" class="nav nav-tabs" role="tablist">
<button class="nav-link active" id="nav-tab-1-1" data-bs-toggle="tab" data-bs-target="#nav-tabpanel-1-1" type="button" role="tab" aria-controls="nav-tabpanel-1-1" aria-selected="true">
  <h3 class="tab-header" id="nav-tab-heading-1-1">
  w/ RNA-seq
  </h3>
</button>
<button class="nav-link" id="nav-tab-1-2" data-bs-toggle="tab" data-bs-target="#nav-tabpanel-1-2" type="button" role="tab" aria-controls="nav-tabpanel-1-2" aria-selected="false">
  <h3 class="tab-header" id="nav-tab-heading-1-2">
  w/ Proteins
  </h3>
</button>
<button class="nav-link" id="nav-tab-1-3" data-bs-toggle="tab" data-bs-target="#nav-tabpanel-1-3" type="button" role="tab" aria-controls="nav-tabpanel-1-3" aria-selected="false">
  <h3 class="tab-header" id="nav-tab-heading-1-3">
  w/ RNA-seq + Proteins
  </h3>
</button>
<button class="nav-link" id="nav-tab-1-4" data-bs-toggle="tab" data-bs-target="#nav-tabpanel-1-4" type="button" role="tab" aria-controls="nav-tabpanel-1-4" aria-selected="false">
  <h3 class="tab-header" id="nav-tab-heading-1-4">
  ab initio
  </h3>
</button>
<button class="nav-link" id="nav-tab-1-5" data-bs-toggle="tab" data-bs-target="#nav-tabpanel-1-5" type="button" role="tab" aria-controls="nav-tabpanel-1-5" aria-selected="false">
  <h3 class="tab-header" id="nav-tab-heading-1-5">
  Pretrained model
  </h3>
</button>
</div>
</nav><div id="nav-tabContent-1" class="tab-content">
<div id="nav-tabpanel-1-1" class="tab-pane show active" aria-labelledby="nav-tab-1-1" role="tabpanel">
<figure><img src="https://github.com/user-attachments/assets/201d283e-83cc-4d2a-bb2b-68104dc72cc3" alt="with RNA-Seq" height="500" class="figure mx-auto d-block"></figure>
</div>
<div id="nav-tabpanel-1-2" class="tab-pane" aria-labelledby="nav-tab-1-2" role="tabpanel">
<figure><img src="https://github.com/user-attachments/assets/6a0cc45f-12f4-4363-81de-93c929166334" alt="with Proteins" height="550" class="figure mx-auto d-block"></figure>
</div>
<div id="nav-tabpanel-1-3" class="tab-pane" aria-labelledby="nav-tab-1-3" role="tabpanel">
<figure><img src="https://github.com/user-attachments/assets/b48adfaa-c357-411d-a252-8450cf720e5f" alt="with RNAseq and proteins" class="figure mx-auto d-block"><div class="figcaption">with RNAseq and proteins</div>
</figure>
</div>
<div id="nav-tabpanel-1-4" class="tab-pane" aria-labelledby="nav-tab-1-4" role="tabpanel">
<figure><img src="https://github.com/user-attachments/assets/5049a336-398a-42b6-834c-3adf75d8be07" alt="_ab initio_" height="500" class="figure mx-auto d-block"></figure>
</div>
<div id="nav-tabpanel-1-5" class="tab-pane" aria-labelledby="nav-tab-1-5" role="tabpanel">
<figure><img src="https://github.com/user-attachments/assets/81dcb3b6-25a1-4c22-a550-f6906a44af4e" alt="_ab initio_" height="500" class="figure mx-auto d-block"></figure>
</div>
</div>
</div>
<p><strong>2. Helixer</strong>: A deep learning-based gene prediction
tool that uses a convolutional neural network (CNN) to predict genes in
eukaryotic genomes.</p>
<ul>
<li>Utilizes a deep learning model trained on diverse animal and plant
genomes, reducing the need for retraining on closely related
species.</li>
<li>Provides probabilities for every base, enabling detailed predictions
for intergenic, untranslated, coding, and intronic regions.</li>
<li>Outperforms traditional tools like AUGUSTUS in both base-wise
metrics and RNA-Seq consistency.</li>
<li>Handles large and complex eukaryotic genomes effectively, even for
species with longer genes.</li>
<li>Requires no additional data except genome – no repeatmasking!</li>
<li>Cannot predict alternative splicing events or spliced start
codons.</li>
</ul>
<figure><img src="https://github.com/user-attachments/assets/fd826e9d-30f9-48fe-9e82-e93cc8d779ce" alt="Helixer" class="figure mx-auto d-block"><div class="figcaption">Helixer</div>
</figure><p><strong>3. EASEL</strong>: Efficient, Accurate, Scalable Eukaryotic
modeLs: eukaryotic genome annotation tool</p>
<ul>
<li>Combines machine learning, RNA folding, and functional annotations
to enhance prediction accuracy.</li>
<li>Leverages RNA-Seq, protein alignments, and additional evidence for
gene prediction.</li>
<li>Built on Nextflow, ensuring portability across HPC systems and
compatibility with Docker and Singularity.</li>
<li>Features EnTAP integration for streamlined functional
annotation.</li>
<li>Provides multiple formats (GFF, GTF, CDS, protein sequences) for
downstream analysis.</li>
<li>Employs BUSCO for quality assessment and automates parameter
optimization.</li>
</ul>
<figure><img src="https://github.com/user-attachments/assets/12e320eb-5ea6-43a9-bedb-2ae64097e44b" alt="EASEL" class="figure mx-auto d-block"><div class="figcaption">EASEL</div>
</figure><p><strong>4. EnTAP</strong>: eukaryotic non-model annotation pipeline
(functional annotation)</p>
<ul>
<li>Optimized for functional annotation of transcriptomes without
reference genomes, addressing challenges like fragmentation and assembly
artifacts.</li>
<li>Much faster than traditional annotation pipelines.</li>
<li>Integrates taxonomic filtering, contaminant detection, and
informativeness metrics for selecting optimal alignments.</li>
<li>Enables annotation across diverse repositories (e.g., RefSeq,
Swiss-Prot) with customizable database integration.</li>
<li>Includes gene family assignments, protein domains, and GO terms for
enriched annotation.</li>
<li>User-friendly, Unix-based pipeline suitable for HPC environments,
combining simplicity with high throughput.</li>
</ul>
<figure><img src="https://github.com/user-attachments/assets/11c2a5e7-2545-45a8-9d59-0b8d233f4de9" alt="EnTAP" class="figure mx-auto d-block"><div class="figcaption">EnTAP</div>
</figure><div id="how-to-we-access-gene-predictions" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="how-to-we-access-gene-predictions" class="callout-inner">
<h3 class="callout-title">How to we access gene predictions?</h3>
<div class="callout-content">
<ul>
<li>
<strong>Busco and Omark</strong> assess gene set completeness by
checking the presence of conserved orthologs across species, helping
identify missing or fragmented genes<br>
</li>
<li>
<strong>Gff3 metrics</strong> provide structural statistics on gene
models, including exon-intron distribution, gene lengths, and coding
sequence properties, highlighting inconsistencies in annotation<br>
</li>
<li>
<strong>Feature assignment</strong> using RNA-seq read mapping
quantifies how well predicted genes capture transcriptomic evidence,
indicating potential missing or incorrect annotations<br>
</li>
<li>
<strong>Reference comparison</strong> with known annotations
measures sensitivity and precision, identifying correctly predicted,
missed, or falsely annotated genes<br>
</li>
<li>
<strong>Multiple validation methods</strong> combine structural,
functional, and comparative assessments to improve confidence in gene
predictions and refine annotation quality</li>
</ul>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Genome annotation strategies include <strong>ab initio,
evidence-based, hybrid, multi-evidence, and large model-based
methods</strong>, each with different strengths and data
requirements.<br>
</li>
<li>
<strong>Ab initio methods</strong> predict genes using sequence
patterns, while <strong>evidence-based methods</strong> use RNA-seq,
proteins, and evolutionary conservation for refinement.<br>
</li>
<li>
<strong>Multi-evidence approaches</strong> integrate diverse
biological data such as ribosome profiling, proteomics, and long-read
RNA-seq to improve accuracy.<br>
</li>
<li>
<strong>Large model-based methods</strong>, like Helixer, use deep
learning to predict genes without requiring manual training,
outperforming traditional tools in complex genomes.<br>
</li>
<li>The workshop focuses on <strong>protein-coding gene
annotation</strong> using tools like <strong>BRAKER, Helixer, EASEL, and
EnTAP</strong>, followed by assessment with BUSCO, OMARK, and structural
metrics.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-annotation-setup"><p>Content from <a href="annotation-setup.html">Annotation Setup</a></p>
<hr>
<p>Last updated on 2025-02-22 |

        <a href="https://rcac-bioinformatics.github.io/genome-annotation/edit/main/episodes/annotation-setup.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How should files and directories be structured for a genome
annotation workflow?<br>
</li>
<li>Why is RNA-seq read mapping important for gene prediction?<br>
</li>
<li>How does repeat masking improve annotation accuracy?<br>
</li>
<li>What preprocessing steps are necessary before running gene
prediction tools?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Set up a structured workspace for genome annotation<br>
</li>
<li>Map RNA-seq reads to the genome to provide evidence for gene
prediction<br>
</li>
<li>Perform repeat masking to prevent false gene annotations<br>
</li>
<li>Prepare genome and transcriptomic data for downstream annotation
analyses</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<hr class="half-width">
<p>This section outlines the preprocessing steps required before gene
annotation, including organizing directories, mapping RNA-Seq reads, and
masking repetitive elements. These steps have already been completed to
save time, but the instructions are provided for reference in case you
need to run them on custom datasets.</p>
<div id="optional-section" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="optional-section" class="callout-inner">
<h3 class="callout-title">Optional Section!</h3>
<div class="callout-content">
<p>Feel free to skip this section as it is already performed to save
time. The steps below are provided for reference - to run on custom
data.</p>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="folder-structure">Folder structure<a class="anchor" aria-label="anchor" href="#folder-structure"></a>
</h2>
<hr class="half-width">
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">${RCAC_SCRATCH}</span>/annotation_workshop</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${RCAC_SCRATCH}</span>/annotation_workshop</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> 00_datasets/<span class="dt">{fastq</span><span class="op">,</span><span class="dt">genome</span><span class="op">,</span><span class="dt">bamfiles}</span></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> 03_setup</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> 04_braker</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> 05_helixer</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> 06_easel</span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> 07_entap</span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> 08_assessment</span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a><span class="fu">rsync</span> <span class="at">-avP</span> SOURCE/00_datasets/ 00_datasets/</span></code></pre>
</div>
<figure><img src="https://github.com/user-attachments/assets/6184ce77-1cf7-49da-86d8-e56a80b29a6e" alt="Organization" class="figure mx-auto d-block"><div class="figcaption">Organization</div>
</figure></section><section><h2 class="section-heading" id="rna-seq-mapping">RNA-Seq mapping<a class="anchor" aria-label="anchor" href="#rna-seq-mapping"></a>
</h2>
<hr class="half-width">
<p>The RNA-seq dataset used for gene prediction comes from the accession
<a href="https://www.ebi.ac.uk/biostudies/arrayexpress/studies/E-MTAB-6422" class="external-link">E-MTAB-6422</a>,
that provides a gene expression atlas of Arabidopsis thaliana (Columbia
accession), capturing transcriptional programs across 36 libraries
representing different tissues (seedling, root, inflorescence, flower,
silique, seed) and developmental stages (2-leaf, 6-leaf, 12-leaf,
senescence, dry mature, and imbibed seed), with three biological
replicates per condition.</p>
<p><strong>Step 1: Download the reference genome and annotation
files</strong></p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="co"># Download the reference genome</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">${WORKSHOP_DIR}</span>/00_datasets/genome</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${WORKSHOP_DIR}</span>/00_datasets/genome </span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="fu">wget</span> https://ftp.ensemblgenomes.ebi.ac.uk/pub/plants/release-60/fasta/arabidopsis_thaliana/dna/Arabidopsis_thaliana.TAIR10.dna.toplevel.fa.gz</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="fu">gunzip</span> Arabidopsis_thaliana.TAIR10.dna.toplevel.fa.gz</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="ex">ml</span> bioawk</span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="ex">bioaw</span> <span class="at">-c</span> fastx <span class="st">'{print "&gt;"$name; print $seq}'</span> <span class="dt">\</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a>   Arabidopsis_thaliana.TAIR10.dna.toplevel.fa <span class="kw">|</span><span class="dt">\</span></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a>   <span class="fu">fold</span> <span class="op">&gt;</span> athaliana.fasta</span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="fu">rm</span> Arabidopsis_thaliana.TAIR10.dna.toplevel.fa</span></code></pre>
</div>
<p><strong>Step 2: Index the reference genome using STAR</strong></p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="ex">ml</span> star</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a><span class="va">genome</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/00_datasets/genome/athaliana.fasta"</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a><span class="va">index</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/00_datasets/genome/star_index"</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a><span class="ex">STAR</span> <span class="at">--runThreadN</span> <span class="va">${SLURM_JOB_CPUS_PER_NODE}</span> <span class="dt">\</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>   <span class="at">--runMode</span> genomeGenerate <span class="dt">\</span></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a>   <span class="at">--genomeDir</span> <span class="va">${index}</span>  <span class="dt">\</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a>   <span class="at">--genomeSAindexNbases</span> 12 <span class="dt">\</span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a>   <span class="at">--genomeFastaFiles</span> <span class="va">${genome}</span></span></code></pre>
</div>
<div id="options" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="options" class="callout-inner">
<h3 class="callout-title">Options</h3>
<div class="callout-content">
<table class="table">
<colgroup>
<col width="27%">
<col width="72%">
</colgroup>
<thead><tr class="header">
<th align="left">Option</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><code>--runThreadN $SLURM_JOB_CPUS_PER_NODE</code></td>
<td>Specifies the number of CPU threads to use. In an HPC environment,
<code>$SLURM_JOB_CPUS_PER_NODE</code> dynamically assigns available CPUs
per node.</td>
</tr>
<tr class="even">
<td align="left"><code>--runMode genomeGenerate</code></td>
<td>Runs STAR in genome index generation mode, required before
alignment.</td>
</tr>
<tr class="odd">
<td align="left"><code>--genomeDir ${index}</code></td>
<td>Path to the directory where the genome index will be stored.</td>
</tr>
<tr class="even">
<td align="left"><code>--genomeSAindexNbases 12</code></td>
<td>Defines the length of the SA pre-indexing string. A recommended
value is <code>min(14, log2(GenomeLength)/2 - 1)</code>, but
<code>12</code> is commonly used for smaller genomes.</td>
</tr>
<tr class="odd">
<td align="left"><code>--genomeFastaFiles ${genome}</code></td>
<td>Path to the FASTA file containing the reference genome
sequences.</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p><strong>Step 3: Map the RNA-Seq reads to the reference
genome</strong></p>
<p>We need to setup a script <code>star-mapping.sh</code> to map the
RNA-Seq reads to the reference genome.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="ex">ml</span> star</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a><span class="va">genome</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/00_datasets/genome/athaliana.fasta"</span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a><span class="va">index</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/00_datasets/genome/star_index"</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a><span class="va">read1</span><span class="op">=</span><span class="va">$1</span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a><span class="va">read2</span><span class="op">=</span><span class="va">$2</span></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a><span class="va">cpus</span><span class="op">=</span><span class="va">${SLURM_JOB_CPUS_PER_NODE}</span></span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a><span class="va">outname</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">${read1}</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 1 <span class="at">-d</span> <span class="st">"_"</span><span class="va">)</span></span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a><span class="ex">STAR</span> <span class="dt">\</span></span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a>--runThreadN <span class="va">${cpus}</span> <span class="dt">\</span></span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a>--genomeDir <span class="va">${index}</span> <span class="dt">\</span></span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a>--outSAMtype BAM SortedByCoordinate <span class="dt">\</span></span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a>--outFilterScoreMinOverLread 0.3 <span class="dt">\</span></span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a>--outFilterMatchNminOverLread 0.3 <span class="dt">\</span></span>
<span id="cb4-18"><a href="#cb4-18" tabindex="-1"></a>--outFileNamePrefix <span class="va">${outname}</span>_ <span class="dt">\</span></span>
<span id="cb4-19"><a href="#cb4-19" tabindex="-1"></a>--readFilesCommand zcat <span class="dt">\</span></span>
<span id="cb4-20"><a href="#cb4-20" tabindex="-1"></a>--outWigType bedGraph <span class="dt">\</span></span>
<span id="cb4-21"><a href="#cb4-21" tabindex="-1"></a>--outWigStrand Unstranded <span class="dt">\</span></span>
<span id="cb4-22"><a href="#cb4-22" tabindex="-1"></a>--outWigNorm RPM <span class="dt">\</span></span>
<span id="cb4-23"><a href="#cb4-23" tabindex="-1"></a>--readFilesIn <span class="va">${read1}</span> <span class="va">${read2}</span></span></code></pre>
</div>
<div id="options-1" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="options-1" class="callout-inner">
<h3 class="callout-title">Options</h3>
<div class="callout-content">
<table class="table">
<colgroup>
<col width="23%">
<col width="76%">
</colgroup>
<thead><tr class="header">
<th align="left">Option</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><code>--runThreadN ${cpus}</code></td>
<td>Specifies the number of CPU threads to use for alignment.</td>
</tr>
<tr class="even">
<td align="left"><code>--genomeDir ${index}</code></td>
<td>Path to the pre-generated STAR genome index directory.</td>
</tr>
<tr class="odd">
<td align="left"><code>--outSAMtype BAM SortedByCoordinate</code></td>
<td>Outputs aligned reads in <strong>BAM format</strong>, sorted by
genomic coordinates.</td>
</tr>
<tr class="even">
<td align="left"><code>--outFilterScoreMinOverLread 0.3</code></td>
<td>Sets the minimum alignment score relative to read length (30% of the
max possible score).</td>
</tr>
<tr class="odd">
<td align="left"><code>--outFilterMatchNminOverLread 0.3</code></td>
<td>Sets the minimum number of matching bases relative to read length
(30% of the read must match).</td>
</tr>
<tr class="even">
<td align="left"><code>--outFileNamePrefix ${outname}_</code></td>
<td>Prefix for all output files generated by STAR.</td>
</tr>
<tr class="odd">
<td align="left"><code>--readFilesCommand zcat</code></td>
<td>Specifies a command (<code>zcat</code>) to decompress input FASTQ
files if they are gzipped.</td>
</tr>
<tr class="even">
<td align="left"><code>--outWigType bedGraph</code></td>
<td>Outputs signal coverage in <strong>bedGraph</strong> format for
visualization.</td>
</tr>
<tr class="odd">
<td align="left"><code>--outWigStrand Unstranded</code></td>
<td>Specifies that the RNA-seq library is <strong>unstranded</strong>
(reads are not strand-specific).</td>
</tr>
<tr class="even">
<td align="left"><code>--outWigNorm RPM</code></td>
<td>Normalizes coverage signal to <strong>Reads Per Million
(RPM)</strong>.</td>
</tr>
<tr class="odd">
<td align="left"><code>--readFilesIn ${read1} ${read2}</code></td>
<td>Specifies input paired-end FASTQ files (Read 1 and Read 2).</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p><strong>Step 4: Run the STAR mapping script</strong></p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="co"># Define the directory containing FASTQ files</span></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="va">FASTQ_DIR</span><span class="op">=</span><span class="st">"WORKSHOP_DIR/00_datasets/fastq"</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="co"># Loop through all R1 files and find corresponding R2 files</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="cf">for</span> R1 <span class="kw">in</span> <span class="va">${FASTQ_DIR}</span>/<span class="pp">*</span>_R1.fq.gz<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>    <span class="va">R2</span><span class="op">=</span><span class="st">"</span><span class="va">${R1</span><span class="op">/</span><span class="ss">_R1.fq.gz</span><span class="op">/</span>_R2.fq.gz<span class="va">}</span><span class="st">"</span>  <span class="co"># Replace R1 with R2</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>    <span class="cf">if</span> <span class="kw">[[</span> <span class="ot">-f</span> <span class="st">"</span><span class="va">$R2</span><span class="st">"</span> <span class="kw">]];</span> <span class="cf">then</span>  <span class="co"># Check if R2 exists</span></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a>        <span class="bu">echo</span> <span class="st">"Running STAR mapping for: </span><span class="va">$R1</span><span class="st"> and </span><span class="va">$R2</span><span class="st">"</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a>        <span class="ex">./star-mapping.sh</span> <span class="st">"</span><span class="va">$R1</span><span class="st">"</span> <span class="st">"</span><span class="va">$R2</span><span class="st">"</span></span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a>    <span class="cf">else</span></span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a>        <span class="bu">echo</span> <span class="st">"Error: Matching R2 file for </span><span class="va">$R1</span><span class="st"> not found!"</span> <span class="op">&gt;&amp;</span><span class="dv">2</span></span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a>    <span class="cf">fi</span></span>
<span id="cb5-14"><a href="#cb5-14" tabindex="-1"></a><span class="cf">done</span></span></code></pre>
</div>
<p><strong>Step 5: store BAM files</strong></p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="co"># Define the directory containing BAM files</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="va">BAM_DIR</span><span class="op">=</span><span class="st">"WORKSHOP_DIR/00_datasets/bamfiles"</span></span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="co"># Move all BAM files to the BAM directory</span></span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a><span class="fu">mv</span> <span class="pp">*</span>_Aligned.sortedByCoord.out.bam <span class="va">${BAM_DIR}</span>/</span></code></pre>
</div>
</section><section><h2 class="section-heading" id="repeatmasker">RepeatMasker<a class="anchor" aria-label="anchor" href="#repeatmasker"></a>
</h2>
<hr class="half-width">
<p>Some gene prediction tools require repeat-masked genomes to avoid
mis-annotation of transposable elements as genes. We will use
RepeatMasker to mask the repeats in the genome. But first, we need to
download the RepeatMasker database.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${WORKSHOP_DIR}</span>/00_datasets/genome</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="co"># Download the RepeatMasker database</span></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="fu">wget</span> https://www.girinst.org/server/RepBase/protected/RepBase30.01.fasta.tar.gz</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="fu">tar</span> xf RepBase30.01.fasta.tar.gz</span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a><span class="va">replib</span><span class="op">=</span><span class="va">$(</span><span class="fu">find</span> <span class="va">$(</span><span class="bu">pwd</span><span class="va">)</span> <span class="at">-name</span> <span class="st">"athrep.ref"</span><span class="va">)</span></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a><span class="ex">ml</span> repeatmasker</span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a><span class="va">genome</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/00_datasets/genome/athaliana.fasta"</span></span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a><span class="ex">RepeatMasker</span> <span class="dt">\</span></span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a>   <span class="at">-e</span> ncbi <span class="dt">\</span></span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a>   <span class="at">-pa</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb7-14"><a href="#cb7-14" tabindex="-1"></a>   <span class="at">-q</span> <span class="dt">\</span></span>
<span id="cb7-15"><a href="#cb7-15" tabindex="-1"></a>   <span class="at">-xsmall</span> <span class="dt">\</span></span>
<span id="cb7-16"><a href="#cb7-16" tabindex="-1"></a>   <span class="at">-lib</span> <span class="va">${replib}</span> <span class="dt">\ </span></span>
<span id="cb7-17"><a href="#cb7-17" tabindex="-1"></a>   <span class="ex">-nocut</span> <span class="dt">\</span></span>
<span id="cb7-18"><a href="#cb7-18" tabindex="-1"></a>   <span class="at">-gff</span> <span class="dt">\</span></span>
<span id="cb7-19"><a href="#cb7-19" tabindex="-1"></a>   <span class="va">${genome}</span></span></code></pre>
</div>
<p>Once done, we will rename it to
<code>athaliana.fasta.masked</code></p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="fu">mv</span> athaliana.fasta.masked athaliana_softmasked.fasta</span></code></pre>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Organizing files and directories ensures a reproducible and
efficient workflow<br>
</li>
<li>Mapping RNA-seq reads to the genome provides essential evidence for
gene prediction<br>
</li>
<li>Repeat masking prevents transposable elements from being
mis-annotated as genes<br>
</li>
<li>Preprocessing steps are crucial for accurate and high-quality genome
annotation</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-gene-annotation-with-braker"><p>Content from <a href="gene-annotation-with-braker.html">Annotation using BRAKER</a></p>
<hr>
<p>Last updated on 2025-02-22 |

        <a href="https://rcac-bioinformatics.github.io/genome-annotation/edit/main/episodes/gene-annotation-with-braker.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What is BRAKER3?</li>
<li>What are the different scenarios in which BRAKER3 can be used?</li>
<li>How to run BRAKER3 with different input requirements?</li>
<li>What are the different output files generated by BRAKER3?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand the different scenarios in which BRAKER3 can be used</li>
<li>Learn how to run BRAKER3 with different input requirements</li>
<li>Learn how to interpret the output files generated by BRAKER3</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>BRAKER3 is a pipeline that combines GeneMark-ET and AUGUSTUS to
predict genes in eukaryotic genomes. This pipeline is particularly
useful for annotating newly sequenced genomes. The flexibility of
BRAKER3 allows users to provide various input datasets for improving
gene prediction accuracy. In this example, we will use various scenarios
to predict genes in a Maize genome using BRAKER3. Following are the
scenarios we will cover</p>
<section><h2 class="section-heading" id="organizing-the-data">Organizing the Data<a class="anchor" aria-label="anchor" href="#organizing-the-data"></a>
</h2>
<hr class="half-width">
<p>Before running BRAKER3, we need to organize the data in the following
format:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">${WORKSHOP_DIR}</span>/04_braker/braker_case{1..5}</span></code></pre>
</div>
<figure><img src="https://github.com/user-attachments/assets/d08a88c4-e847-4e5d-b3cc-c60db10453d6" alt="Folder organization" class="figure mx-auto d-block"><div class="figcaption">Folder organization</div>
</figure></section><section><h2 class="section-heading" id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<hr class="half-width">
<p>Before running BRAKER3, we need to set up:</p>
<ol style="list-style-type: decimal">
<li>
<code>GeneMark-ES/ET/EP/ETP</code> license key</li>
<li>The <code>AUGUSTUS_CONFIG_PATH</code> configuration path</li>
</ol>
<p>The license key for <code>GeneMark-ES/ET/EP/ETP</code> can be
obtained from the <a href="http://exon.gatech.edu/GeneMark/license_download.cgi" class="external-link">GeneMark
website</a>. Once downloaded, you need to place it in your home
directory:</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">tar</span> xf gm_key_64.gz</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">cp</span> gm_key_64 ~/.gm_key</span></code></pre>
</div>
<p>For the <code>AUGUSTUS_CONFIG_PATH</code>, we need to copy the
<code>config</code> directory from the Singularity container to the
scratch directory. This is required because BRAKER3 needs to write to
the <code>config</code> directory, and the Singularity container is
read-only. To copy the <code>config</code> directory, run the following
command:</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="ex">ml</span> braker3/v3.0.7.5</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">${WORKSHOP_DIR}</span>/04_braker/augustus</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a><span class="ex">copy_augustus_config</span> <span class="va">${WORKSHOP_DIR}</span>/04_braker/augustus</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a><span class="bu">export</span> <span class="va">AUGUSTUS_CONFIG_PATH</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/04_braker/augustus/config"</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="case-1-using-rnaseq-reads">Case 1: Using RNAseq Reads<a class="anchor" aria-label="anchor" href="#case-1-using-rnaseq-reads"></a>
</h2>
<hr class="half-width">
<p>To run BRAKER3 with RNAseq reads, we need to provide the RNAseq reads
in BAM format. We will use the following command to run BRAKER3 with
RNAseq reads:</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="co">#SBATCH --nodes=1</span></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="co">#SBATCH --ntasks=32</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="co">#SBATCH --account=workshop</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="co">#SBATCH --time=04-00:00:00</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a><span class="co">#SBATCH --job-name=braker_case1</span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a><span class="co">#SBATCH --output=%x.o%j</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a><span class="co">#SBATCH --error=%x.e%j</span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a><span class="co"># Load the required modules</span></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a><span class="ex">ml</span> braker3/v3.0.7.5</span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a><span class="co"># directory/file paths</span></span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a><span class="va">BAM_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/00_datasets/bamfiles"</span></span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a><span class="va">BAM_FILES</span><span class="op">=</span><span class="va">$(</span><span class="fu">find</span> <span class="st">"</span><span class="va">${BAM_DIR}</span><span class="st">"</span> <span class="at">-name</span> <span class="st">"*.bam"</span> <span class="kw">|</span> <span class="fu">tr</span> <span class="st">'\n'</span> <span class="st">','</span><span class="va">)</span></span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a><span class="va">genome</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/00_datasets/genome/athaliana_softmasked.fasta"</span></span>
<span id="cb4-18"><a href="#cb4-18" tabindex="-1"></a><span class="co"># runtime parameters</span></span>
<span id="cb4-19"><a href="#cb4-19" tabindex="-1"></a><span class="va">AUGUSTUS_CONFIG_PATH</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/04_braker/augustus/config"</span></span>
<span id="cb4-20"><a href="#cb4-20" tabindex="-1"></a><span class="va">GENEMARK_PATH</span><span class="op">=</span><span class="st">"/opt/ETP/bin/gmes"</span></span>
<span id="cb4-21"><a href="#cb4-21" tabindex="-1"></a><span class="co"># Run BRAKER3</span></span>
<span id="cb4-22"><a href="#cb4-22" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/04_braker/braker_case1</span>
<span id="cb4-23"><a href="#cb4-23" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${workdir}</span></span>
<span id="cb4-24"><a href="#cb4-24" tabindex="-1"></a><span class="ex">braker.pl</span> <span class="dt">\</span></span>
<span id="cb4-25"><a href="#cb4-25" tabindex="-1"></a>   <span class="at">--AUGUSTUS_CONFIG_PATH</span><span class="op">=</span><span class="va">${AUGUSTUS_CONFIG_PATH}</span> <span class="dt">\</span></span>
<span id="cb4-26"><a href="#cb4-26" tabindex="-1"></a>   <span class="at">--GENEMARK_PATH</span><span class="op">=</span><span class="va">${GENEMARK_PATH}</span> <span class="dt">\</span></span>
<span id="cb4-27"><a href="#cb4-27" tabindex="-1"></a>   <span class="at">--bam</span><span class="op">=</span><span class="va">${BAM_FILES}</span> <span class="dt">\</span></span>
<span id="cb4-28"><a href="#cb4-28" tabindex="-1"></a>   <span class="at">--genome</span><span class="op">=</span><span class="va">${genome}</span> <span class="dt">\</span></span>
<span id="cb4-29"><a href="#cb4-29" tabindex="-1"></a>   <span class="at">--species</span><span class="op">=</span>At_<span class="va">$(</span><span class="fu">date</span> +<span class="st">"%Y%m%d"</span><span class="va">)</span>.c1b <span class="dt">\</span></span>
<span id="cb4-30"><a href="#cb4-30" tabindex="-1"></a>   <span class="at">--workingdir</span><span class="op">=</span><span class="va">${workdir}</span> <span class="dt">\</span></span>
<span id="cb4-31"><a href="#cb4-31" tabindex="-1"></a>   <span class="at">--gff3</span> <span class="dt">\</span></span>
<span id="cb4-32"><a href="#cb4-32" tabindex="-1"></a>   <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span></code></pre>
</div>
<p>The job takes ~55 mins to finish. The results of the BRAKER3 run will
be stored in the respective <code>braker_case1</code> directory, as
<code>braker.gff3</code> file.</p>
</section><section><h2 class="section-heading" id="case-2-using-conserved-proteins-only">Case 2: Using Conserved proteins only<a class="anchor" aria-label="anchor" href="#case-2-using-conserved-proteins-only"></a>
</h2>
<hr class="half-width">
<p>Using the <a href="https://github.com/tomasbruna/orthodb-clades" class="external-link">orthodb-clades</a>
tool, we can download protein sequences for a specific clade. In this
scenario, since we are using the Arabidopsis genome, we can download the
<code>clades</code> specific <code>Viridiplantae.fa</code> <a href="https://www.orthodb.org/" class="external-link">OrthoDB v12</a> protein sets.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/04_braker/braker_case2</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">${workdir}</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="fu">git</span> clone git@github.com:tomasbruna/orthodb-clades.git</span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="bu">cd</span> orthodb-clades</span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a><span class="ex">ml</span> snakemake</span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a><span class="ex">snakemake</span> <span class="at">--cores</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> selectViridiplantae</span></code></pre>
</div>
<p>When this is done, you should see a folder named <code>clade</code>
with <code>Viridiplantae.fa</code> in the <code>orthodb-clades</code>
directory. We will use this as one of the input datasets for
BRAKER3.<br>
The following command will run BRAKER3 with the input genome and protein
sequences:</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="co">#SBATCH --nodes=1</span></span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="co">#SBATCH --ntasks=32</span></span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a><span class="co">#SBATCH --time=04-00:00:00</span></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a><span class="co">#SBATCH --job-name=braker_case2</span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a><span class="co">#SBATCH --output=%x.o%j</span></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a><span class="co">#SBATCH --error=%x.e%j</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a><span class="co"># Load the required modules</span></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a><span class="ex">ml</span> braker3/v3.0.7.5</span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a><span class="co"># directory/file paths</span></span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a><span class="va">proteins</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/04_braker/braker_case2/orthodb-clades/clades/Viridiplantae.fa"</span></span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a><span class="va">genome</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/00_datasets/genome/athaliana_softmasked.fasta"</span></span>
<span id="cb6-16"><a href="#cb6-16" tabindex="-1"></a><span class="co"># runtime parameters</span></span>
<span id="cb6-17"><a href="#cb6-17" tabindex="-1"></a><span class="va">AUGUSTUS_CONFIG_PATH</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/04_braker/augustus/config"</span></span>
<span id="cb6-18"><a href="#cb6-18" tabindex="-1"></a><span class="va">GENEMARK_PATH</span><span class="op">=</span><span class="st">"/opt/ETP/bin/gmes"</span></span>
<span id="cb6-19"><a href="#cb6-19" tabindex="-1"></a><span class="co"># Run BRAKER3</span></span>
<span id="cb6-20"><a href="#cb6-20" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/04_braker/braker_case2</span>
<span id="cb6-21"><a href="#cb6-21" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">${workdir}</span></span>
<span id="cb6-22"><a href="#cb6-22" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${workdir}</span></span>
<span id="cb6-23"><a href="#cb6-23" tabindex="-1"></a><span class="ex">braker.pl</span> <span class="dt">\</span></span>
<span id="cb6-24"><a href="#cb6-24" tabindex="-1"></a>   <span class="at">--AUGUSTUS_CONFIG_PATH</span><span class="op">=</span><span class="va">${AUGUSTUS_CONFIG_PATH}</span> <span class="dt">\</span></span>
<span id="cb6-25"><a href="#cb6-25" tabindex="-1"></a>   <span class="at">--GENEMARK_PATH</span><span class="op">=</span><span class="va">${GENEMARK_PATH}</span> <span class="dt">\</span></span>
<span id="cb6-26"><a href="#cb6-26" tabindex="-1"></a>   <span class="at">--genome</span><span class="op">=</span><span class="va">${genome}</span> <span class="dt">\</span></span>
<span id="cb6-27"><a href="#cb6-27" tabindex="-1"></a>   <span class="at">--prot_seq</span><span class="op">=</span><span class="va">${proteins}</span> <span class="dt">\</span></span>
<span id="cb6-28"><a href="#cb6-28" tabindex="-1"></a>   <span class="at">--species</span><span class="op">=</span>At_<span class="va">$(</span><span class="fu">date</span> +<span class="st">"%Y%m%d"</span><span class="va">)</span>.c2 <span class="dt">\</span></span>
<span id="cb6-29"><a href="#cb6-29" tabindex="-1"></a>   <span class="at">--workingdir</span><span class="op">=</span><span class="va">${workdir}</span> <span class="dt">\</span></span>
<span id="cb6-30"><a href="#cb6-30" tabindex="-1"></a>   <span class="at">--gff3</span> <span class="dt">\</span></span>
<span id="cb6-31"><a href="#cb6-31" tabindex="-1"></a>   <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span></code></pre>
</div>
<p>The results of the BRAKER3 run will be stored in the respective
<code>braker_case2</code> directory, as <code>braker.gff3</code> file.
The run time for this job is ~170 mins.</p>
</section><section><h2 class="section-heading" id="case-3-using-rnaseq-and-proteins-together">Case 3: Using RNAseq and Proteins Together<a class="anchor" aria-label="anchor" href="#case-3-using-rnaseq-and-proteins-together"></a>
</h2>
<hr class="half-width">
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="co">#SBATCH --nodes=1</span></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="co">#SBATCH --ntasks=32</span></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="co">#SBATCH --time=04-00:00:00</span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="co">#SBATCH --job-name=braker_case3</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a><span class="co">#SBATCH --output=%x.o%j</span></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a><span class="co">#SBATCH --error=%x.e%j</span></span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a><span class="co"># Load the required modules</span></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a><span class="ex">ml</span> braker3/v3.0.7.5</span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a><span class="co"># directory/file paths</span></span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb7-14"><a href="#cb7-14" tabindex="-1"></a><span class="va">BAM_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/00_datasets/bamfiles"</span></span>
<span id="cb7-15"><a href="#cb7-15" tabindex="-1"></a><span class="va">BAM_FILES</span><span class="op">=</span><span class="va">$(</span><span class="fu">find</span> <span class="st">"</span><span class="va">${BAM_DIR}</span><span class="st">"</span> <span class="at">-name</span> <span class="st">"*.bam"</span> <span class="kw">|</span> <span class="fu">tr</span> <span class="st">'\n'</span> <span class="st">','</span><span class="va">)</span></span>
<span id="cb7-16"><a href="#cb7-16" tabindex="-1"></a><span class="va">proteins</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/04_braker/braker_case2/orthodb-clades/clade/Viridiplantae.fa"</span></span>
<span id="cb7-17"><a href="#cb7-17" tabindex="-1"></a><span class="va">genome</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/00_datasets/genome/athaliana_softmasked.fasta"</span></span>
<span id="cb7-18"><a href="#cb7-18" tabindex="-1"></a><span class="co"># runtime parameters</span></span>
<span id="cb7-19"><a href="#cb7-19" tabindex="-1"></a><span class="va">AUGUSTUS_CONFIG_PATH</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/04_braker/augustus/config"</span></span>
<span id="cb7-20"><a href="#cb7-20" tabindex="-1"></a><span class="va">GENEMARK_PATH</span><span class="op">=</span><span class="st">"/opt/ETP/bin/gmes"</span></span>
<span id="cb7-21"><a href="#cb7-21" tabindex="-1"></a><span class="co"># Run BRAKER3</span></span>
<span id="cb7-22"><a href="#cb7-22" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/04_braker/braker_case3</span>
<span id="cb7-23"><a href="#cb7-23" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">${workdir}</span></span>
<span id="cb7-24"><a href="#cb7-24" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${workdir}</span></span>
<span id="cb7-25"><a href="#cb7-25" tabindex="-1"></a><span class="ex">braker.pl</span> <span class="dt">\</span></span>
<span id="cb7-26"><a href="#cb7-26" tabindex="-1"></a>   <span class="at">--AUGUSTUS_CONFIG_PATH</span><span class="op">=</span><span class="va">${AUGUSTUS_CONFIG_PATH}</span> <span class="dt">\</span></span>
<span id="cb7-27"><a href="#cb7-27" tabindex="-1"></a>   <span class="at">--GENEMARK_PATH</span><span class="op">=</span><span class="va">${GENEMARK_PATH}</span> <span class="dt">\</span></span>
<span id="cb7-28"><a href="#cb7-28" tabindex="-1"></a>   <span class="at">--genome</span><span class="op">=</span><span class="va">${genome}</span> <span class="dt">\</span></span>
<span id="cb7-29"><a href="#cb7-29" tabindex="-1"></a>   <span class="at">--prot_seq</span><span class="op">=</span><span class="va">${proteins}</span> <span class="dt">\</span></span>
<span id="cb7-30"><a href="#cb7-30" tabindex="-1"></a>   <span class="at">--bam</span><span class="op">=</span><span class="va">${BAM_FILES}</span> <span class="dt">\</span></span>
<span id="cb7-31"><a href="#cb7-31" tabindex="-1"></a>   <span class="at">--species</span><span class="op">=</span>At_<span class="va">$(</span><span class="fu">date</span> +<span class="st">"%Y%m%d"</span><span class="va">)</span>.c3 <span class="dt">\</span></span>
<span id="cb7-32"><a href="#cb7-32" tabindex="-1"></a>   <span class="at">--workingdir</span><span class="op">=</span><span class="va">${workdir}</span> <span class="dt">\</span></span>
<span id="cb7-33"><a href="#cb7-33" tabindex="-1"></a>   <span class="at">--gff3</span> <span class="dt">\</span></span>
<span id="cb7-34"><a href="#cb7-34" tabindex="-1"></a>   <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span></code></pre>
</div>
<p>The results of the BRAKER3 run will be stored in the respective
<code>braker_case3</code> directory, as <code>braker.gff3</code> file.
The run time for this job is ~120 mins.</p>
</section><section><h2 class="section-heading" id="case-4-ab-initio-mode">Case 4: <em>ab Initio</em> mode<a class="anchor" aria-label="anchor" href="#case-4-ab-initio-mode"></a>
</h2>
<hr class="half-width">
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="co">#SBATCH --nodes=1</span></span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="co">#SBATCH --ntasks=32</span></span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a><span class="co">#SBATCH --time=04-00:00:00</span></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a><span class="co">#SBATCH --job-name=braker_case4</span></span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a><span class="co">#SBATCH --output=%x.o%j</span></span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a><span class="co">#SBATCH --error=%x.e%j</span></span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a><span class="co"># Load the required modules</span></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a><span class="ex">ml</span> braker3/v3.0.7.5</span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a><span class="co"># directory/file paths</span></span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a><span class="va">genome</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/00_datasets/genome/athaliana_softmasked.fasta"</span></span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a><span class="co"># runtime parameters</span></span>
<span id="cb8-16"><a href="#cb8-16" tabindex="-1"></a><span class="va">AUGUSTUS_CONFIG_PATH</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/04_braker/augustus/config"</span></span>
<span id="cb8-17"><a href="#cb8-17" tabindex="-1"></a><span class="va">GENEMARK_PATH</span><span class="op">=</span><span class="st">"/opt/ETP/bin/gmes"</span></span>
<span id="cb8-18"><a href="#cb8-18" tabindex="-1"></a><span class="co"># Run BRAKER3</span></span>
<span id="cb8-19"><a href="#cb8-19" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/04_braker/braker_case4</span>
<span id="cb8-20"><a href="#cb8-20" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">${workdir}</span></span>
<span id="cb8-21"><a href="#cb8-21" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${workdir}</span></span>
<span id="cb8-22"><a href="#cb8-22" tabindex="-1"></a><span class="ex">braker.pl</span> <span class="dt">\</span></span>
<span id="cb8-23"><a href="#cb8-23" tabindex="-1"></a>   <span class="at">--AUGUSTUS_CONFIG_PATH</span><span class="op">=</span><span class="va">${AUGUSTUS_CONFIG_PATH}</span> <span class="dt">\</span></span>
<span id="cb8-24"><a href="#cb8-24" tabindex="-1"></a>   <span class="at">--GENEMARK_PATH</span><span class="op">=</span><span class="va">${GENEMARK_PATH}</span> <span class="dt">\</span></span>
<span id="cb8-25"><a href="#cb8-25" tabindex="-1"></a>   <span class="at">--esmode</span> <span class="dt">\</span></span>
<span id="cb8-26"><a href="#cb8-26" tabindex="-1"></a>   <span class="at">--genome</span><span class="op">=</span><span class="va">${genome}</span> <span class="dt">\</span></span>
<span id="cb8-27"><a href="#cb8-27" tabindex="-1"></a>   <span class="at">--species</span><span class="op">=</span>Zm_<span class="va">$(</span><span class="fu">date</span> +<span class="st">"%Y%m%d"</span><span class="va">)</span>.c4 <span class="dt">\</span></span>
<span id="cb8-28"><a href="#cb8-28" tabindex="-1"></a>   <span class="at">--workingdir</span><span class="op">=</span><span class="va">${workdir}</span> <span class="dt">\</span></span>
<span id="cb8-29"><a href="#cb8-29" tabindex="-1"></a>   <span class="at">--gff3</span> <span class="dt">\</span></span>
<span id="cb8-30"><a href="#cb8-30" tabindex="-1"></a>   <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span></code></pre>
</div>
<p>The results of the BRAKER3 run will be stored in the respective
<code>braker_case4</code> directory, as <code>braker.gff3</code> file.
The run time for this job is ~60 mins.</p>
</section><section><h2 class="section-heading" id="case-5-using-pre-trained-model">Case 5: Using Pre-trained Model<a class="anchor" aria-label="anchor" href="#case-5-using-pre-trained-model"></a>
</h2>
<hr class="half-width">
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="co">#SBATCH --nodes=1</span></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a><span class="co">#SBATCH --ntasks=32</span></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a><span class="co">#SBATCH --time=04-00:00:00</span></span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a><span class="co">#SBATCH --job-name=braker_case5</span></span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a><span class="co">#SBATCH --output=%x.o%j</span></span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a><span class="co">#SBATCH --error=%x.e%j</span></span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a><span class="co"># Load the required modules</span></span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a><span class="ex">ml</span> braker3/v3.0.7.5</span>
<span id="cb9-12"><a href="#cb9-12" tabindex="-1"></a><span class="co"># directory/file paths</span></span>
<span id="cb9-13"><a href="#cb9-13" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb9-14"><a href="#cb9-14" tabindex="-1"></a><span class="va">genome</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/00_datasets/genome/athaliana_softmasked.fasta"</span></span>
<span id="cb9-15"><a href="#cb9-15" tabindex="-1"></a><span class="co"># runtime parameters</span></span>
<span id="cb9-16"><a href="#cb9-16" tabindex="-1"></a><span class="va">AUGUSTUS_CONFIG_PATH</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/04_braker/augustus/config"</span></span>
<span id="cb9-17"><a href="#cb9-17" tabindex="-1"></a><span class="va">GENEMARK_PATH</span><span class="op">=</span><span class="st">"/opt/ETP/bin/gmes"</span></span>
<span id="cb9-18"><a href="#cb9-18" tabindex="-1"></a><span class="co"># Run BRAKER3</span></span>
<span id="cb9-19"><a href="#cb9-19" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/04_braker/braker_case5</span>
<span id="cb9-20"><a href="#cb9-20" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">${workdir}</span></span>
<span id="cb9-21"><a href="#cb9-21" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${workdir}</span></span>
<span id="cb9-22"><a href="#cb9-22" tabindex="-1"></a><span class="ex">braker.pl</span> <span class="dt">\</span></span>
<span id="cb9-23"><a href="#cb9-23" tabindex="-1"></a>   <span class="at">--AUGUSTUS_CONFIG_PATH</span><span class="op">=</span><span class="va">${AUGUSTUS_CONFIG_PATH}</span> <span class="dt">\</span></span>
<span id="cb9-24"><a href="#cb9-24" tabindex="-1"></a>   <span class="at">--GENEMARK_PATH</span><span class="op">=</span><span class="va">${GENEMARK_PATH}</span> <span class="dt">\</span></span>
<span id="cb9-25"><a href="#cb9-25" tabindex="-1"></a>   <span class="at">--skipAllTraining</span> <span class="dt">\</span></span>
<span id="cb9-26"><a href="#cb9-26" tabindex="-1"></a>   <span class="at">--genome</span><span class="op">=</span><span class="va">${genome}</span> <span class="dt">\</span></span>
<span id="cb9-27"><a href="#cb9-27" tabindex="-1"></a>   <span class="at">--species</span><span class="op">=</span>arabidopsis <span class="dt">\</span></span>
<span id="cb9-28"><a href="#cb9-28" tabindex="-1"></a>   <span class="at">--workingdir</span><span class="op">=</span><span class="va">${workdir}</span> <span class="dt">\</span></span>
<span id="cb9-29"><a href="#cb9-29" tabindex="-1"></a>   <span class="at">--gff3</span> <span class="dt">\</span></span>
<span id="cb9-30"><a href="#cb9-30" tabindex="-1"></a>   <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span></code></pre>
</div>
<p>The results of the BRAKER3 run will be stored in the respective
<code>braker_case5/Augustus</code> directory. The main result file
<code>augustus.ab_initio.gff3</code> will be used for downstream
analysis. Runtime for this job is ~7 mins.</p>
</section><section><h2 class="section-heading" id="results-and-outputs">Results and Outputs<a class="anchor" aria-label="anchor" href="#results-and-outputs"></a>
</h2>
<hr class="half-width">
<p>coming soon!</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>BRAKER3 is a pipeline that combines GeneMark-ET and AUGUSTUS to
predict genes in eukaryotic genomes</li>
<li>BRAKER3 can be used with different input datasets to improve gene
prediction accuracy</li>
<li>BRAKER3 can be run in different scenarios to predict genes in a
genome</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-gene-annotation-with-helixer"><p>Content from <a href="gene-annotation-with-helixer.html">Annotation using Helixer</a></p>
<hr>
<p>Last updated on 2025-02-22 |

        <a href="https://rcac-bioinformatics.github.io/genome-annotation/edit/main/episodes/gene-annotation-with-helixer.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How to predict genes using Helixer?</li>
<li>How to download trained models for Helixer?</li>
<li>How to run Helixer on the HPC cluster (Gilbreth)?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Learn how to predict genes using Helixer.</li>
<li>Learn how to download trained models for Helixer.</li>
<li>Learn how to run Helixer on the HPC cluster (Gilbreth).</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>Helixer is a deep learning-based gene prediction tool that uses a
convolutional neural network (CNN) to predict genes in eukaryotic
genomes. Helixer is trained on a wide range of eukaryotic genomes and
can predict genes in both plant and animal genomes. Helixer can predict
genes wihtout any extrinisic information such as RNA-seq data or
homology information, purely based on the sequence of the genome.</p>
<div id="this-section-should-be-run-on-gilbreth-hpc-cluster." class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="this-section-should-be-run-on-gilbreth-hpc-cluster." class="callout-inner">
<h3 class="callout-title">This section should be run on Gilbreth HPC cluster.</h3>
<div class="callout-content">
<p>Due to the GPU requirement for Helixer, you need to run this section
on the Gilbreth HPC cluster. You don’t have to copy fastq, or bamfiles,
but only need <code>athaliana.fasta</code> file in the
<code>00_datasets/genome</code> directory.</p>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">${WORKSHOP_DIR}</span>/05_helixer</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">cp</span> <span class="op">&lt;</span>depot_location<span class="op">&gt;</span>/genome/athaliana.fasta <span class="va">${WORKSHOP_DIR}</span>/05_helixer/athaliana.fasta</span></code></pre>
</div>
<figure><img src="https://github.com/user-attachments/assets/a1767236-304b-4912-8200-e2ea97ec63b8" alt="Folder organization" class="figure mx-auto d-block"><div class="figcaption">Folder organization</div>
</figure><section><h2 class="section-heading" id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<hr class="half-width">
<p>Helixer is available as a Singularity/apptainer container. You can
pull the container using the following <code>apptainer pull</code>
command. See the <a href="https://github.com/gglyptodon/helixer-docker" class="external-link">Helixer Docker</a>
repository for more information.</p>
<p>Helixer is installed as a module on the Gilbreth cluster, and can be
loaded using the following commands:</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="ex">ml</span> helixer</span></code></pre>
</div>
</section><section><h2 class="section-heading" id="downloading-trained-models">Downloading trained models<a class="anchor" aria-label="anchor" href="#downloading-trained-models"></a>
</h2>
<hr class="half-width">
<p>Helixer requires a trained model to predict genes. With the included
script <code>fetch_helixer_models.py</code> you can download models for
specific lineages. Currently, models are available for the following
lineages:</p>
<ul>
<li><code>land_plant</code></li>
<li><code>vertebrate</code></li>
<li><code>invertibrate</code></li>
<li><code>fungi</code></li>
</ul>
<p>You can download the models using the following command:</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="co"># all models</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="co"># fetch_helixer_models.py --all</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="co"># or for a specific lineage</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="ex">fetch_helixer_models.py</span> <span class="at">--lineage</span> land_plant</span></code></pre>
</div>
<p>This will download all lineage models in the models directory. You
can also download models for specific lineages using the
<code>--lineage</code> option as shown above.</p>
<p>By default, files will be downloaded to
<code>~/.local/share/Helixer</code> directory. You should see the
follwing files:</p>
<pre><code>.
└── models
    └── land_plant
        ├── land_plant_v0.3_a_0080.h5
        ├── land_plant_v0.3_a_0090.h5
        ├── land_plant_v0.3_a_0100.h5
        ├── land_plant_v0.3_a_0200.h5
        ├── land_plant_v0.3_a_0300.h5
        ├── land_plant_v0.3_a_0400.h5
        ├── land_plant_v0.3_m_0100.h5
        └── land_plant_v0.3_m_0200.h5</code></pre>
<p><code>land_plant_v0.3_a_0080.h5</code> is the smallest model and
<code>land_plant_v0.3_m_0200.h5</code> is the largest model. The model
size is determined by the number of parameters in the model. The larger
models are more accurate but require more memory and time to run.</p>
</section><section><h2 class="section-heading" id="running-helixer">Running Helixer<a class="anchor" aria-label="anchor" href="#running-helixer"></a>
</h2>
<hr class="half-width">
<p>Helixer requires GPU for prediction. For running Helixer, you need to
request a GPU node. You will also need the genome sequence in fasta
format. For this tutorial, we will use Maize genome (Zea mays subsp.
mays), and use the <code>land_plant</code> model to predict genes.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="co">#SBATCH --nodes=1</span></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="co">#SBATCH --ntasks-per-node=16</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="co">#SBATCH --gpus-per-node=1</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="co">#SBATCH --time=04:00:00</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="co">#SBATCH --account=standby</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="co">#SBATCH --job-name=helixer</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a><span class="co">#SBATCH --output=%x.o%j</span></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a><span class="co">#SBATCH --error=%x.e%j</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a><span class="ex">ml</span> helixer</span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a><span class="va">genome</span><span class="op">=</span>athaliana.fasta</span>
<span id="cb5-14"><a href="#cb5-14" tabindex="-1"></a><span class="va">species</span><span class="op">=</span><span class="st">"Arabidopsis thaliana"</span></span>
<span id="cb5-15"><a href="#cb5-15" tabindex="-1"></a><span class="va">output</span><span class="op">=</span><span class="st">"athaliana_helixer.gff"</span></span>
<span id="cb5-16"><a href="#cb5-16" tabindex="-1"></a><span class="ex">Helixer.py</span> <span class="dt">\</span></span>
<span id="cb5-17"><a href="#cb5-17" tabindex="-1"></a>    <span class="at">--lineage</span> land_plant <span class="dt">\</span></span>
<span id="cb5-18"><a href="#cb5-18" tabindex="-1"></a>    <span class="at">--fasta-path</span> <span class="va">${genome}</span> <span class="dt">\</span></span>
<span id="cb5-19"><a href="#cb5-19" tabindex="-1"></a>    <span class="at">--species</span> <span class="va">${species}</span> <span class="dt">\</span></span>
<span id="cb5-20"><a href="#cb5-20" tabindex="-1"></a>    <span class="at">--gff-output-path</span> <span class="va">${output}</span></span></code></pre>
</div>
<p>A typical job will take ~40 mins to finish depedning on the GPU the
job gets allocated.</p>
<p>You can count the nubmer of predictions in your gff3 file using the
following command:</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">awk</span> <span class="st">'$3=="gene"'</span> athaliana_helixer.gff <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span></code></pre>
</div>
<p>The GFF format output had 27,201 genes predicted using Helixer. You
can view the various features in the gff file using the following
command:</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-v</span> <span class="st">"^#"</span> athaliana_helixer.gff <span class="kw">|</span><span class="dt">\</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>   <span class="fu">cut</span> <span class="at">-f</span> 3 <span class="kw">|</span><span class="dt">\</span></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>   <span class="fu">sort</span> <span class="kw">|</span><span class="dt">\</span></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>   <span class="fu">uniq</span> <span class="at">-c</span></span></code></pre>
</div>
<p>To get <code>cds</code> and <code>pep</code> files, you can use the
following command:</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="ex">ml</span> cufflinks</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a><span class="ex">gffread</span> Arabidopsis_thaliana.gff3 <span class="dt">\</span></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>   <span class="at">-g</span> athaliana.fasta <span class="dt">\</span></span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a>   <span class="at">-y</span> helixer_pep.fa <span class="dt">\</span></span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>   <span class="at">-x</span> helixer_cds.fa</span></code></pre>
</div>
<p>As you may have noticed, the number of mRNA and gene features are the
same. This is because isoforms aren’t predicted by Helixer and you only
have one transcript per gene. Exons are indetified with high confidence
and alternative isoforms are usually collapsed into a single gene model.
This is one of the known limitations of Helixer.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Helixer is a deep learning-based gene prediction tool that uses a
convolutional neural network (CNN) to predict genes in eukaryotic
genomes.</li>
<li>Helixer can predict genes wihtout any extrinisic information such as
RNA-seq data or homology information, purely based on the sequence of
the genome.</li>
<li>Helixer requires a trained model and GPU for prediction.</li>
<li>Helixer predicts genes in the GFF3 format, but will not predict
isoforms.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-gene-annotation-with-easel"><p>Content from <a href="gene-annotation-with-easel.html">Annotation using Easel</a></p>
<hr>
<p>Last updated on 2025-02-22 |

        <a href="https://rcac-bioinformatics.github.io/genome-annotation/edit/main/episodes/gene-annotation-with-easel.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What are the steps required to set up and run EASEL on an HPC
system?</li>
<li>How is Nextflow used to manage and execute the EASEL workflow?</li>
<li>What configuration files need to be modified before running
EASEL?</li>
<li>How do you submit and monitor an EASEL job using Slurm?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Set up and configure EASEL for execution on an HPC system.</li>
<li>Install Nextflow and pull the EASEL workflow from the
repository.</li>
<li>Modify the necessary configuration files to match the HPC
environment.</li>
<li>Submit and run EASEL as a Slurm job and verify successful
execution.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>EASEL (Efficient, Accurate, Scalable Eukaryotic modeLs) is a genome
annotation tool that integrates machine learning, RNA folding, and
functional annotations to improve gene prediction accuracy, leveraging
optimized AUGUSTUS parameters, transcriptome and protein evidence, and a
Nextflow-based scalable pipeline for reproducible analysis.</p>
<section><h2 class="section-heading" id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<hr class="half-width">
<p>We will do a custom installation of nextflow (we will not use
<code>ml nextflow</code>):</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="ex">ml</span> openjdk</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-s</span> https://get.nextflow.io <span class="kw">|</span> <span class="fu">bash</span></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="fu">mv</span> nextflow ~/.local/bin</span></code></pre>
</div>
<p>check the installation:</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">nextflow</span> <span class="at">-v</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">which</span> nextflow</span></code></pre>
</div>
<p>We can organize our folder structure as follows:</p>
<figure><img src="https://github.com/user-attachments/assets/ce52353b-f0c9-4fae-977f-58b2efa26bd1" alt="Folder organization" class="figure mx-auto d-block"><div class="figcaption">Folder organization</div>
</figure></section><section><h2 class="section-heading" id="running-easel">Running Easel<a class="anchor" aria-label="anchor" href="#running-easel"></a>
</h2>
<hr class="half-width">
<p><strong>Step 1:</strong> create a <code>rcac.config</code> file to
run the jobs using <code>slurm</code> scheduler.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="ex">params</span> {</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>    <span class="ex">config_profile_description</span> = <span class="st">'Negishi HPC, RCAC Purdue University.'</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>    <span class="ex">config_profile_contact</span>     = <span class="st">'Arun Seetharam (aseethar@purdue.edu)'</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>    <span class="ex">config_profile_url</span>         = <span class="st">"https://www.rcac.purdue.edu/knowledge/negishi"</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>    <span class="ex">partition</span>                  = <span class="st">'testpbs'</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>    <span class="ex">project</span>                    = <span class="st">'your_project_name'</span>  // Replace with a valid Slurm project</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>    <span class="ex">max_memory</span>                 = <span class="st">'256GB'</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>    <span class="ex">max_cpus</span>                   = 128</span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>    <span class="ex">max_time</span>                   = <span class="st">'48h'</span></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a><span class="er">}</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a><span class="ex">singularity</span> {</span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a>    <span class="ex">enabled</span>    = true</span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a>    <span class="ex">autoMounts</span> = true</span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a>    <span class="ex">singularity.cacheDir</span> = <span class="st">"</span><span class="va">${System</span><span class="er">.getenv('RCAC_SCRATCH')</span><span class="va">}</span><span class="st">/.singularity"</span></span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a>    <span class="ex">singularity.pullTimeout</span> = <span class="st">"1h"</span></span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a><span class="er">}</span></span>
<span id="cb3-17"><a href="#cb3-17" tabindex="-1"></a><span class="ex">process</span> {</span>
<span id="cb3-18"><a href="#cb3-18" tabindex="-1"></a>    <span class="ex">resourceLimits</span> = [</span>
<span id="cb3-19"><a href="#cb3-19" tabindex="-1"></a>        <span class="ex">memory:</span> 256.GB,</span>
<span id="cb3-20"><a href="#cb3-20" tabindex="-1"></a>        <span class="ex">cpus:</span> 128,</span>
<span id="cb3-21"><a href="#cb3-21" tabindex="-1"></a>        <span class="ex">time:</span> 48.h</span>
<span id="cb3-22"><a href="#cb3-22" tabindex="-1"></a>    <span class="ex">]</span></span>
<span id="cb3-23"><a href="#cb3-23" tabindex="-1"></a>    <span class="ex">executor</span>       = <span class="st">'slurm'</span></span>
<span id="cb3-24"><a href="#cb3-24" tabindex="-1"></a>    <span class="ex">clusterOptions</span> = <span class="st">"-A </span><span class="va">${params</span><span class="er">.project</span><span class="va">}</span><span class="st"> -p </span><span class="va">${params</span><span class="er">.partition</span><span class="va">}</span><span class="st">"</span></span>
<span id="cb3-25"><a href="#cb3-25" tabindex="-1"></a><span class="er">}</span></span>
<span id="cb3-26"><a href="#cb3-26" tabindex="-1"></a><span class="ex">//</span> Executor-specific configurations</span>
<span id="cb3-27"><a href="#cb3-27" tabindex="-1"></a><span class="cf">if</span> <span class="kw">(</span><span class="ex">params.executor</span> == <span class="st">'slurm'</span><span class="kw">)</span> <span class="kw">{</span></span>
<span id="cb3-28"><a href="#cb3-28" tabindex="-1"></a>    <span class="ex">process</span> {</span>
<span id="cb3-29"><a href="#cb3-29" tabindex="-1"></a>        <span class="ex">cpus</span>   = { check_max<span class="er">(</span> <span class="ex">16</span>    <span class="pp">*</span> task.attempt, <span class="st">'cpus'</span>   <span class="kw">)</span> <span class="kw">}</span></span>
<span id="cb3-30"><a href="#cb3-30" tabindex="-1"></a>        <span class="ex">memory</span> = { check_max<span class="er">(</span> <span class="ex">32.GB</span> <span class="pp">*</span> task.attempt, <span class="st">'memory'</span> <span class="kw">)</span> <span class="er">}</span></span>
<span id="cb3-31"><a href="#cb3-31" tabindex="-1"></a>        <span class="bu">time</span>   = { check_max<span class="er">(</span> <span class="ex">8.h</span>  <span class="pp">*</span> task.attempt, <span class="st">'time'</span>   <span class="kw">)</span> <span class="er">}</span></span>
<span id="cb3-32"><a href="#cb3-32" tabindex="-1"></a>        <span class="ex">clusterOptions</span> = <span class="st">"-A </span><span class="va">${params</span><span class="er">.project</span><span class="va">}</span><span class="st"> -p </span><span class="va">${params</span><span class="er">.partition</span><span class="va">}</span><span class="st">"</span></span>
<span id="cb3-33"><a href="#cb3-33" tabindex="-1"></a></span>
<span id="cb3-34"><a href="#cb3-34" tabindex="-1"></a>        <span class="ex">errorStrategy</span> = { task.exitStatus in <span class="pp">[</span><span class="ss">143,137,104,134,139</span><span class="pp">]</span> <span class="pp">?</span> <span class="st">'retry'</span> : <span class="st">'finish'</span> }</span>
<span id="cb3-35"><a href="#cb3-35" tabindex="-1"></a>        <span class="ex">maxRetries</span>    = 1</span>
<span id="cb3-36"><a href="#cb3-36" tabindex="-1"></a>        <span class="ex">maxErrors</span>     = <span class="st">'-1'</span></span>
<span id="cb3-37"><a href="#cb3-37" tabindex="-1"></a></span>
<span id="cb3-38"><a href="#cb3-38" tabindex="-1"></a>        <span class="ex">withLabel:process_single</span> {</span>
<span id="cb3-39"><a href="#cb3-39" tabindex="-1"></a>            <span class="ex">cpus</span>   = { check_max<span class="er">(</span> <span class="ex">16</span>                  , <span class="st">'cpus'</span>    <span class="kw">)</span> <span class="er">}</span></span>
<span id="cb3-40"><a href="#cb3-40" tabindex="-1"></a>            <span class="ex">memory</span> = { check_max<span class="er">(</span> <span class="ex">32.GB</span>  <span class="pp">*</span> task.attempt, <span class="st">'memory'</span>  <span class="kw">)</span> <span class="er">}</span></span>
<span id="cb3-41"><a href="#cb3-41" tabindex="-1"></a>            <span class="bu">time</span>   = { check_max<span class="er">(</span> <span class="ex">8.h</span>    <span class="pp">*</span> task.attempt, <span class="st">'time'</span>    <span class="kw">)</span> <span class="er">}</span></span>
<span id="cb3-42"><a href="#cb3-42" tabindex="-1"></a>        <span class="er">}</span></span>
<span id="cb3-43"><a href="#cb3-43" tabindex="-1"></a>        <span class="ex">withLabel:process_low</span> {</span>
<span id="cb3-44"><a href="#cb3-44" tabindex="-1"></a>            <span class="ex">cpus</span>   = { check_max<span class="er">(</span> <span class="ex">32</span>     <span class="pp">*</span> task.attempt, <span class="st">'cpus'</span>    <span class="kw">)</span> <span class="er">}</span></span>
<span id="cb3-45"><a href="#cb3-45" tabindex="-1"></a>            <span class="ex">memory</span> = { check_max<span class="er">(</span> <span class="ex">64.GB</span>  <span class="pp">*</span> task.attempt, <span class="st">'memory'</span>  <span class="kw">)</span> <span class="er">}</span></span>
<span id="cb3-46"><a href="#cb3-46" tabindex="-1"></a>            <span class="bu">time</span>   = { check_max<span class="er">(</span> <span class="ex">24.h</span>   <span class="pp">*</span> task.attempt, <span class="st">'time'</span>    <span class="kw">)</span> <span class="er">}</span></span>
<span id="cb3-47"><a href="#cb3-47" tabindex="-1"></a>        <span class="er">}</span></span>
<span id="cb3-48"><a href="#cb3-48" tabindex="-1"></a>        <span class="ex">withLabel:process_medium</span> {</span>
<span id="cb3-49"><a href="#cb3-49" tabindex="-1"></a>            <span class="ex">cpus</span>   = { check_max<span class="er">(</span> <span class="ex">64</span>     <span class="pp">*</span> task.attempt, <span class="st">'cpus'</span>    <span class="kw">)</span> <span class="er">}</span></span>
<span id="cb3-50"><a href="#cb3-50" tabindex="-1"></a>            <span class="ex">memory</span> = { check_max<span class="er">(</span> <span class="ex">128.GB</span> <span class="pp">*</span> task.attempt, <span class="st">'memory'</span>  <span class="kw">)</span> <span class="er">}</span></span>
<span id="cb3-51"><a href="#cb3-51" tabindex="-1"></a>            <span class="bu">time</span>   = { check_max<span class="er">(</span> <span class="ex">24.h</span>   <span class="pp">*</span> task.attempt, <span class="st">'time'</span>    <span class="kw">)</span> <span class="er">}</span></span>
<span id="cb3-52"><a href="#cb3-52" tabindex="-1"></a>        <span class="er">}</span></span>
<span id="cb3-53"><a href="#cb3-53" tabindex="-1"></a>        <span class="ex">withLabel:process_high</span> {</span>
<span id="cb3-54"><a href="#cb3-54" tabindex="-1"></a>            <span class="ex">cpus</span>   = { check_max<span class="er">(</span> <span class="ex">128</span>    <span class="pp">*</span> task.attempt, <span class="st">'cpus'</span>    <span class="kw">)</span> <span class="er">}</span></span>
<span id="cb3-55"><a href="#cb3-55" tabindex="-1"></a>            <span class="ex">memory</span> = { check_max<span class="er">(</span> <span class="ex">256.GB</span> <span class="pp">*</span> task.attempt, <span class="st">'memory'</span>  <span class="kw">)</span> <span class="er">}</span></span>
<span id="cb3-56"><a href="#cb3-56" tabindex="-1"></a>            <span class="bu">time</span>   = { check_max<span class="er">(</span> <span class="ex">24.h</span>   <span class="pp">*</span> task.attempt, <span class="st">'time'</span>    <span class="kw">)</span> <span class="er">}</span></span>
<span id="cb3-57"><a href="#cb3-57" tabindex="-1"></a>        <span class="er">}</span></span>
<span id="cb3-58"><a href="#cb3-58" tabindex="-1"></a>        <span class="ex">withLabel:process_long</span> {</span>
<span id="cb3-59"><a href="#cb3-59" tabindex="-1"></a>            <span class="bu">time</span>   = { check_max<span class="er">(</span> <span class="ex">48.h</span>  <span class="pp">*</span> task.attempt, <span class="st">'time'</span>    <span class="kw">)</span> <span class="er">}</span></span>
<span id="cb3-60"><a href="#cb3-60" tabindex="-1"></a>        <span class="er">}</span></span>
<span id="cb3-61"><a href="#cb3-61" tabindex="-1"></a>        <span class="ex">withLabel:process_high_memory</span> {</span>
<span id="cb3-62"><a href="#cb3-62" tabindex="-1"></a>            <span class="ex">memory</span> = { check_max<span class="er">(</span> <span class="ex">256.GB</span> <span class="pp">*</span> task.attempt, <span class="st">'memory'</span> <span class="kw">)</span> <span class="er">}</span></span>
<span id="cb3-63"><a href="#cb3-63" tabindex="-1"></a>        <span class="er">}</span></span>
<span id="cb3-64"><a href="#cb3-64" tabindex="-1"></a>        <span class="ex">withLabel:error_ignore</span> {</span>
<span id="cb3-65"><a href="#cb3-65" tabindex="-1"></a>            <span class="ex">errorStrategy</span> = <span class="st">'ignore'</span></span>
<span id="cb3-66"><a href="#cb3-66" tabindex="-1"></a>        <span class="er">}</span></span>
<span id="cb3-67"><a href="#cb3-67" tabindex="-1"></a>        <span class="ex">withLabel:error_retry</span> {</span>
<span id="cb3-68"><a href="#cb3-68" tabindex="-1"></a>            <span class="ex">errorStrategy</span> = <span class="st">'retry'</span></span>
<span id="cb3-69"><a href="#cb3-69" tabindex="-1"></a>            <span class="ex">maxRetries</span>    = 2</span>
<span id="cb3-70"><a href="#cb3-70" tabindex="-1"></a>        <span class="er">}</span></span>
<span id="cb3-71"><a href="#cb3-71" tabindex="-1"></a>    <span class="er">}</span></span>
<span id="cb3-72"><a href="#cb3-72" tabindex="-1"></a><span class="er">}</span></span></code></pre>
</div>
<p><strong>Step 2:</strong> pull the Easel nextflow workflow:</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="ex">ml</span> nextflow</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="ex">ml</span> openjdk</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="ex">nextflow</span> pull <span class="at">-hub</span> gitlab PlantGenomicsLab/easel</span></code></pre>
</div>
<p>This will pull the latest version of the Easel workflow from the
PlantGenomicsLab GitLab repository.</p>
<p><strong>Step 3:</strong> modify the
<code>~/.nextflow/assets/PlantGenomicsLab/easel/nextflow.config</code>
file to include the <code>rcac.config</code> file:</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="ex">vim</span> ~/.nextflow/assets/PlantGenomicsLab/easel/nextflow.config</span></code></pre>
</div>
<p>or if you prefer <code>nano</code>:</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">nano</span> ~/.nextflow/assets/PlantGenomicsLab/easel/nextflow.config</span></code></pre>
</div>
<p>Add the following line to the <code>nextflow.config</code> file:</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="ex">profiles</span> {</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>   <span class="ex">debug</span> { process.beforeScript = <span class="st">'echo $HOSTNAME'</span> }</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>   <span class="ex">...</span></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>   <span class="ex">...</span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>   <span class="bu">test</span>  { includeConfig <span class="st">'conf/test.config'</span> }</span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>   <span class="ex">rcac</span>  { includeConfig <span class="st">'conf/rcac.config'</span> }</span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a><span class="er">}</span></span></code></pre>
</div>
<p>Note that this is in line # 235 in the <code>nextflow.config</code>
file. and all the lines between <code>debug</code> and <code>test</code>
aren’t shown here.</p>
<p>Now, we are ready to run the Easel workflow.</p>
<p><strong>Step 4:</strong> create <code>param.yaml</code> file to run
the Easel workflow:</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">${WORKSHOP_DIR}</span>/06_easel</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${WORKSHOP_DIR}</span>/06_easel</span></code></pre>
</div>
<p>for the <code>param.yaml</code> file:</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">YAML<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode yaml" tabindex="0"><code class="sourceCode yaml"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="fu">outdir</span><span class="kw">:</span><span class="at"> easel</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="fu">genome</span><span class="kw">:</span><span class="at"> /scratch/negishi/aseethar/annotation_workshop/00_datasets/genome/athaliana_softmasked.fasta</span></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a><span class="fu">bam</span><span class="kw">:</span><span class="at"> /scratch/negishi/aseethar/annotation_workshop/00_datasets/bamfiles/*.bam</span></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a><span class="fu">busco_lineage</span><span class="kw">:</span><span class="at"> embryophyta</span></span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a><span class="fu">order</span><span class="kw">:</span><span class="at"> Brassicales</span></span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a><span class="fu">prefix</span><span class="kw">:</span><span class="at"> arabidopsis</span></span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a><span class="fu">taxon</span><span class="kw">:</span><span class="at"> arabidopsis</span></span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a><span class="fu">singularity_cache_dir</span><span class="kw">:</span><span class="at"> /scratch/negishi/aseethar/singularity_cache</span></span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a><span class="fu">training_set</span><span class="kw">:</span><span class="at"> plant</span></span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a><span class="fu">executor</span><span class="kw">:</span><span class="at"> slurm</span></span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a><span class="fu">account</span><span class="kw">:</span><span class="at"> testpbs</span></span>
<span id="cb9-12"><a href="#cb9-12" tabindex="-1"></a><span class="fu">qos</span><span class="kw">:</span><span class="at"> normal</span></span>
<span id="cb9-13"><a href="#cb9-13" tabindex="-1"></a><span class="fu">project</span><span class="kw">:</span><span class="at"> testpbs</span></span></code></pre>
</div>
<p>Now we are ready to run the Easel workflow:</p>
<p>Our <code>slurm</code> file to run the Easel workflow:</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a><span class="co">#SBATCH --nodes=1</span></span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a><span class="co">#SBATCH --ntasks=8</span></span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a><span class="co">#SBATCH --time=04-00:00:00</span></span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a><span class="co">#SBATCH --job-name=easel</span></span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a><span class="co">#SBATCH --output=%x.o%j</span></span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a><span class="co">#SBATCH --error=%x.e%j</span></span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a><span class="co"># Load the required modules</span></span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a><span class="ex">ml</span> purge</span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb10-11"><a href="#cb10-11" tabindex="-1"></a><span class="ex">ml</span> openjdk</span>
<span id="cb10-12"><a href="#cb10-12" tabindex="-1"></a><span class="ex">nextflow</span> run <span class="dt">\</span></span>
<span id="cb10-13"><a href="#cb10-13" tabindex="-1"></a>   <span class="at">-hub</span> gitlab PlantGenomicsLab/easel <span class="dt">\</span></span>
<span id="cb10-14"><a href="#cb10-14" tabindex="-1"></a>   <span class="at">-params-file</span> params.yaml <span class="dt">\</span></span>
<span id="cb10-15"><a href="#cb10-15" tabindex="-1"></a>   <span class="at">-profile</span> rcac <span class="dt">\</span></span>
<span id="cb10-16"><a href="#cb10-16" tabindex="-1"></a>   <span class="at">--project</span> testpbs</span></code></pre>
</div>
<p>This will run the Easel workflow on the RCAC HPC.</p>
</section><section><h2 class="section-heading" id="results-and-outputs">Results and Outputs<a class="anchor" aria-label="anchor" href="#results-and-outputs"></a>
</h2>
<hr class="half-width">
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>EASEL is executed using Nextflow, which simplifies workflow
management and ensures reproducibility.</li>
<li>Proper configuration of resource settings and HPC parameters is
essential for successful job execution.</li>
<li>Running EASEL requires setting up input files, modifying
configuration files, and submitting jobs via Slurm.</li>
<li>Understanding how to monitor and troubleshoot jobs helps ensure
efficient pipeline execution.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-functional-annotation-with-entap"><p>Content from <a href="functional-annotation-with-entap.html">Functional annotation using EnTAP</a></p>
<hr>
<p>Last updated on 2025-02-22 |

        <a href="https://rcac-bioinformatics.github.io/genome-annotation/edit/main/episodes/functional-annotation-with-entap.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What is EnTAP, and how does it improve functional annotation for
non-model transcriptomes?</li>
<li>How does EnTAP filter, annotate, and assign functional roles to
predicted transcripts?</li>
<li>What databases and evidence sources does EnTAP integrate for
annotation?</li>
<li>What are the key steps required to set up and execute EnTAP on an
HPC system?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand how EnTAP improves functional annotation for non-model
eukaryotes.</li>
<li>Learn how EnTAP processes transcript data through filtering,
alignment, and functional assignment.</li>
<li>Set up and modify EnTAP configuration files for correct
execution.</li>
<li>Run EnTAP on an HPC system and interpret the generated
annotations.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="entap-overview">EnTAP Overview<a class="anchor" aria-label="anchor" href="#entap-overview"></a>
</h2>
<hr class="half-width">
<p>EnTAP is a bioinformatics pipeline designed to enhance the accuracy,
speed, and flexibility of functional annotation for de novo assembled
transcriptomes in non-model eukaryotic organisms. It mitigates assembly
fragmentation, improves annotation rates, and provides extensive
functional insights into transcriptomes. You can provide predicted
transcripts in FASTA format and a GFF file containing gene models to run
EnTAP.</p>
<div id="key-features" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="key-features" class="callout-inner">
<h3 class="callout-title">Key Features</h3>
<div class="callout-content">
<ul>
<li>
<strong>Optimized for non-model eukaryotes</strong> – Overcomes
challenges of fragmented transcriptome assemblies.</li>
<li>
<strong>Fast &amp; efficient</strong> – Runs significantly faster
than comparable annotation tools.</li>
<li>
<strong>Customizable</strong> – Supports optional filtering and
analysis steps for user-specific needs.</li>
<li>
<strong>Comprehensive functional insights</strong> – Integrates
multiple annotation sources for high-confidence gene assignments.</li>
<li>
<strong>Contaminant detection</strong> – Helps remove misleading
sequences for cleaner datasets.</li>
</ul>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="how-entap-works">How EnTAP Works<a class="anchor" aria-label="anchor" href="#how-entap-works"></a>
</h2>
<hr class="half-width">
<ol style="list-style-type: decimal">
<li>
<strong>Transcriptome Filtering</strong> – Identifies true coding
sequences (CDS) and removes assembly artifacts:
<ul>
<li>
<strong>Expression Filtering (optional)</strong> – Filters
transcripts based on gene expression levels using RSEM.</li>
<li>
<strong>Frame Selection (optional)</strong> – Further refines CDS
predictions using TransDecoder.</li>
</ul>
</li>
<li>
<strong>Transcriptome Annotation</strong> – Assigns functional
information to sequences:
<ul>
<li>
<strong>Similarity Search</strong> – Rapid alignment against
user-selected databases using DIAMOND.</li>
<li>
<strong>Contaminant Filtering &amp; Best Hit Selection</strong> –
Identifies optimal annotations and flags potential contaminants.</li>
<li>
<strong>Orthologous Group Assignment</strong> – Assigns translated
proteins to gene families using <strong>eggNOG/eggnog-mapper</strong>,
including:
<ul>
<li>
<strong>Protein Domains</strong> (SMART/Pfam)</li>
<li><strong>Gene Ontology (GO) Terms</strong></li>
<li><strong>KEGG Pathway Annotation</strong></li>
</ul>
</li>
<li>
<strong>InterProScan (optional)</strong> – Searches InterPro
databases for additional domain, GO, and pathway annotations.</li>
<li>
<strong>Horizontal Gene Transfer Analysis (optional)</strong> –
Detects potential horizontal gene transfer (HGT) events via
DIAMOND.</li>
</ul>
</li>
</ol></section><section><h2 class="section-heading" id="running-entap">Running EnTAP<a class="anchor" aria-label="anchor" href="#running-entap"></a>
</h2>
<hr class="half-width">
<p>EnTAP is available as a module on the HPC cluster. You can load the
module using the following commands:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="ex">ml</span> entap</span></code></pre>
</div>
<div id="first-time-setup" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="first-time-setup" class="callout-inner">
<h3 class="callout-title">First-Time Setup</h3>
<div class="callout-content">
<p>When running for the first time, you will have to set up the
databases for EnTAP. This includes downloading files from various
databases, and can be time consuming. This section is already performed
so you can skip this step and is included for reference.</p>
</div>
</div>
</div>
<div id="accordionSpoiler1" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler1" aria-expanded="false" aria-controls="collapseSpoiler1">
  <h3 class="accordion-header" id="headingSpoiler1">
  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>
  Setting up databases (optional)
  </h3>
</button>
<div id="collapseSpoiler1" class="accordion-collapse collapse" data-bs-parent="#accordionSpoiler1" aria-labelledby="headingSpoiler1">
<div class="accordion-body">
<p><code>entap_run.params</code> file should be setup as follows (be
sure to select the correct <code>databases</code> for your
organism):</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">INI<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode ini" tabindex="0"><code class="sourceCode ini"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="dt">out-dir</span><span class="ot">=</span><span class="st">entap_dbfiles</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="dt">overwrite</span><span class="ot">=</span><span class="kw">false</span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="dt">resume</span><span class="ot">=</span><span class="kw">false</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="dt">input</span><span class="ot">=</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="dt">database</span><span class="ot">=</span><span class="st">uniprot_sprot,refseq_plant</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="dt">no-trim</span><span class="ot">=</span><span class="kw">false</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="dt">threads</span><span class="ot">=</span><span class="dv">1</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="dt">output-format</span><span class="ot">=</span><span class="st">1,3,4,7,</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="dt">fpkm</span><span class="ot">=</span><span class="fl">0.5</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="dt">align</span><span class="ot">=</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a><span class="dt">single-end</span><span class="ot">=</span><span class="kw">false</span></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a><span class="dt">frame-selection</span><span class="ot">=</span><span class="kw">false</span></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="dt">transdecoder-m</span><span class="ot">=</span><span class="dv">100</span></span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a><span class="dt">transdecoder-no-refine-starts</span><span class="ot">=</span><span class="kw">false</span></span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a><span class="dt">taxon</span><span class="ot">=</span></span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a><span class="dt">qcoverage</span><span class="ot">=</span><span class="dv">50</span></span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a><span class="dt">tcoverage</span><span class="ot">=</span><span class="dv">50</span></span>
<span id="cb2-18"><a href="#cb2-18" tabindex="-1"></a><span class="dt">contam</span><span class="ot">=</span></span>
<span id="cb2-19"><a href="#cb2-19" tabindex="-1"></a><span class="dt">e-value</span><span class="ot">=</span><span class="st">1e-05</span></span>
<span id="cb2-20"><a href="#cb2-20" tabindex="-1"></a><span class="dt">uninformative</span><span class="ot">=</span><span class="st">conserved,predicted,unknown,unnamed,hypothetical,putative,unidentified,uncharacterized,uncultured,uninformative,</span></span>
<span id="cb2-21"><a href="#cb2-21" tabindex="-1"></a><span class="dt">diamond-sensitivity</span><span class="ot">=</span><span class="st">very-sensitive</span></span>
<span id="cb2-22"><a href="#cb2-22" tabindex="-1"></a><span class="dt">ontology_source</span><span class="ot">=</span><span class="st">0,</span></span>
<span id="cb2-23"><a href="#cb2-23" tabindex="-1"></a><span class="dt">eggnog-contaminant</span><span class="ot">=</span><span class="kw">true</span></span>
<span id="cb2-24"><a href="#cb2-24" tabindex="-1"></a><span class="dt">eggnog-dbmem</span><span class="ot">=</span><span class="kw">true</span></span>
<span id="cb2-25"><a href="#cb2-25" tabindex="-1"></a><span class="dt">eggnog-sensitivity</span><span class="ot">=</span><span class="st">more-sensitive</span></span>
<span id="cb2-26"><a href="#cb2-26" tabindex="-1"></a><span class="dt">interproscan-db</span><span class="ot">=</span></span>
<span id="cb2-27"><a href="#cb2-27" tabindex="-1"></a><span class="dt">hgt-donor</span><span class="ot">=</span></span>
<span id="cb2-28"><a href="#cb2-28" tabindex="-1"></a><span class="dt">hgt-recipient</span><span class="ot">=</span></span>
<span id="cb2-29"><a href="#cb2-29" tabindex="-1"></a><span class="dt">hgt-gff</span><span class="ot">=</span></span>
<span id="cb2-30"><a href="#cb2-30" tabindex="-1"></a><span class="dt">ncbi-api-key</span><span class="ot">=</span></span>
<span id="cb2-31"><a href="#cb2-31" tabindex="-1"></a><span class="dt">ncbi-api-enable</span><span class="ot">=</span><span class="kw">true</span></span></code></pre>
</div>
<p><code>entap_config.ini</code> file should be setup as follows (be
sure to modify the paths <code>&lt;custom_location&gt;</code> to your
desired location):</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">INI<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode ini" tabindex="0"><code class="sourceCode ini"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="dt">data-generate</span><span class="ot">=</span><span class="kw">false</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="dt">data-type</span><span class="ot">=</span><span class="st">0,</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="dt">entap-db-bin</span><span class="ot">=</span><span class="st">&lt;custom_location&gt;/entap_db/bin/entap_database.bin</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="dt">entap-db-sql</span><span class="ot">=</span><span class="st">entap_database.db</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="dt">entap-graph</span><span class="ot">=</span><span class="st">entap_graphing.py</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a><span class="dt">rsem-calculate-expression</span><span class="ot">=</span><span class="st">rsem-calculate-expression</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a><span class="dt">rsem-sam-validator</span><span class="ot">=</span><span class="st">rsem-sam-validator</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a><span class="dt">rsem-prepare-reference</span><span class="ot">=</span><span class="st">rsem-prepare-reference</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a><span class="dt">convert-sam-for-rsem</span><span class="ot">=</span><span class="st">convert-sam-for-rsem</span></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a><span class="dt">transdecoder-long-exe</span><span class="ot">=</span><span class="st">TransDecoder.LongOrfs</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a><span class="dt">transdecoder-predict-exe</span><span class="ot">=</span><span class="st">TransDecoder.Predict</span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a><span class="dt">diamond-exe</span><span class="ot">=</span><span class="st">diamond</span></span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a><span class="dt">eggnog-map-exe</span><span class="ot">=</span><span class="st">emapper.py</span></span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a><span class="dt">eggnog-map-data</span><span class="ot">=</span><span class="st">&lt;custom_location&gt;/entap_db/databases</span></span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a><span class="dt">eggnog-map-dmnd</span><span class="ot">=</span><span class="st">&lt;custom_location&gt;/entap_db/bin/eggnog_proteins.dmnd</span></span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a><span class="dt">interproscan-exe</span><span class="ot">=</span><span class="st">interproscan.sh</span></span></code></pre>
</div>
<p>Once done, for the first time setup, you can run the following
command:</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="ex">ml</span> entap</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="ex">EnTAP</span> <span class="dt">\</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>   <span class="at">--config</span> <span class="dt">\</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>   <span class="at">--run-ini</span> ./entap_run.params <span class="dt">\</span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a>   <span class="at">--entap-ini</span> ./entap_config.ini <span class="dt">\</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a>   <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span></code></pre>
</div>
<p>This will download the databases and set up the configuration files
for EnTAP.</p>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="step-1-prepare-files">
<strong>Step 1:</strong> Prepare files<a class="anchor" aria-label="anchor" href="#step-1-prepare-files"></a>
</h2>
<hr class="half-width">
<p>Your input files should be in the following format:</p>
<ul>
<li>
<strong>Transcript FASTA file</strong> – Contains predicted
transcripts in FASTA format.</li>
<li>
<strong>Configuration file</strong> – Specifies parameters for EnTAP
execution.
<ul>
<li>
<code>entap_run.params</code> – Contains runtime parameters for
EnTAP.</li>
<li>
<code>entap_config.ini</code> – Specifies paths to EnTAP binaries
and databases. (you can copy the files from
<code>/depot/itap/datasets/entap_db/entap_{config.ini,run.params}</code>)</li>
</ul>
</li>
</ul>
<p>Edit the <code>entap_run.params</code> file to specify the output
directory for EnTAP results and the correct input file</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">INI<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode ini" tabindex="0"><code class="sourceCode ini"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="dt">out-dir</span><span class="ot">=</span><span class="st">entap_out</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="dt">input</span><span class="ot">=</span><span class="st">input_cds.fasta</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="step-2-run-entap">
<strong>Step 2:</strong> Run EnTAP<a class="anchor" aria-label="anchor" href="#step-2-run-entap"></a>
</h2>
<hr class="half-width">
<p>Run EnTAP using the following command:</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="ex">ml</span> entap</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a><span class="ex">EnTAP</span> <span class="at">-</span><span class="dt">\</span></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>   <span class="at">-run</span> <span class="dt">\</span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>   <span class="at">--run-ini</span> ./entap_run.params <span class="dt">\</span></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a>   <span class="at">--entap-ini</span> ./entap_config.ini <span class="dt">\</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a>   <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="interpreting-results">Interpreting Results<a class="anchor" aria-label="anchor" href="#interpreting-results"></a>
</h2>
<hr class="half-width">
<p>EnTAP generates several output files, but the key results will be in
the <code>entap_out/final_results</code> directory.</p>
<p><strong>EnTAP Output Files Summary</strong></p>
<table class="table">
<colgroup>
<col width="42%">
<col width="57%">
</colgroup>
<thead><tr class="header">
<th align="left"><strong>File/Directory</strong></th>
<th><strong>Description</strong></th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><strong>Final Results
(<code>final_results/</code>)</strong></td>
<td></td>
</tr>
<tr class="even">
<td align="left"><code>annotated.fnn</code></td>
<td>FASTA file of <strong>annotated</strong> transcripts.</td>
</tr>
<tr class="odd">
<td align="left"><code>annotated.tsv</code></td>
<td>Tab-separated file with <strong>functional
annotations</strong>.</td>
</tr>
<tr class="even">
<td align="left"><code>annotated_gene_ontology_terms.tsv</code></td>
<td>GO terms assigned to annotated transcripts.</td>
</tr>
<tr class="odd">
<td align="left"><code>entap_results.tsv</code></td>
<td>
<strong>Master summary</strong> of all results, including
annotations.</td>
</tr>
<tr class="even">
<td align="left"><code>unannotated.fnn</code></td>
<td>FASTA file of <strong>unannotated</strong> transcripts.</td>
</tr>
<tr class="odd">
<td align="left"><code>unannotated.tsv</code></td>
<td>List of transcripts that <strong>failed</strong> annotation.</td>
</tr>
<tr class="even">
<td align="left"><strong><code>gene_family/</code></strong></td>
<td>Stores <strong>eggNOG</strong> gene family assignments, including
orthologs and functional annotations.</td>
</tr>
<tr class="odd">
<td align="left"><strong><code>similarity_search/</code></strong></td>
<td>Contains results from <strong>DIAMOND BLASTX</strong> searches
against selected databases.</td>
</tr>
<tr class="even">
<td align="left"><strong><code>transcriptomes/</code></strong></td>
<td>Holds the <strong>input transcriptome (CDS)</strong> and the
<strong>processed</strong> version after filtering.</td>
</tr>
</tbody>
</table>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>EnTAP enhances functional annotation by integrating multiple
evidence sources, including homology, protein domains, and gene
ontology.</li>
<li>Proper setup of configuration files and databases is essential for
accurate and efficient EnTAP execution.</li>
<li>Running EnTAP involves transcript filtering, similarity searches,
and functional annotation through automated workflows.</li>
<li>The pipeline provides extensive insights into transcript function,
improving downstream biological interpretations.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-annotation-assesment"><p>Content from <a href="annotation-assesment.html">Annotation Assesment</a></p>
<hr>
<p>Last updated on 2025-02-22 |

        <a href="https://rcac-bioinformatics.github.io/genome-annotation/edit/main/episodes/annotation-assesment.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How to assess the quality of a genome annotation?</li>
<li>What are the different tools available for assessing the quality of
a genome annotation?</li>
<li>How to compare the predicted annotation with the reference
annotation?</li>
<li>How to measure the number of raw reads assigned to the features
predicted by the annotation?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Assess the quality of a genome annotation using different
tools.</li>
<li>Compare the predicted annotation with the reference annotation.</li>
<li>Measure the number of raw reads assigned to the features predicted
by the annotation.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<hr class="half-width">
<p>The folder organization is as follows:</p>
<p>The annotation files (<code>gff3</code>) are from the previous steps.
But are collected in this folder for convenience.</p>
<p>We will extract CDS and Protein sequences from the <code>gff3</code>
file before we begin the assesment.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="ex">ml</span> cufflinks</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/08_assessment</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${workdir}</span></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> <span class="pp">*</span>.gff3<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a>    <span class="va">base</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">${i</span><span class="op">%</span>.<span class="pp">*</span><span class="va">})</span></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a>    <span class="ex">gffread</span> <span class="dt">\</span></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a>       <span class="at">-g</span> <span class="va">${WORKSHOP_DIR}</span>/00_datasets/genome/athaliana.fa <span class="dt">\</span></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a>       <span class="at">-x</span> <span class="va">${base}</span>_cds.fa <span class="dt">\</span></span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a>       <span class="at">-y</span> <span class="va">${base}</span>_pep.fa <span class="va">$i</span></span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a><span class="cf">done</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="gff3-metrics">GFF3 metrics<a class="anchor" aria-label="anchor" href="#gff3-metrics"></a>
</h2>
<hr class="half-width">
<p>We will use <code>agat_sp_statistics.pl</code> to calculate the
statistics of the GFF3 file.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="ex">ml</span> agat</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/08_assessment</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${workdir}</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> gff3_stats</span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="cf">for</span> gff3 <span class="kw">in</span> <span class="pp">*</span>.gff3<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a>    <span class="va">base</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">$gff3</span> .gff3<span class="va">)</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a>    <span class="ex">agat_sp_statistics.pl</span> <span class="dt">\</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a>       <span class="at">--gff</span> <span class="va">${gff3}</span> <span class="dt">\</span></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a>       <span class="at">-o</span> gff3_stats/<span class="va">${base}</span>_stats.txt</span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="cf">done</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="busco-assesment">BUSCO assesment<a class="anchor" aria-label="anchor" href="#busco-assesment"></a>
</h2>
<hr class="half-width">
<p>We will use BUSCO to asses the quality of the annotation.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="ex">ml</span> busco</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/08_assessment</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${workdir}</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a><span class="cf">for</span> pep <span class="kw">in</span> <span class="pp">*</span>_pep.fa<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>    <span class="va">base</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">${pep</span><span class="op">%</span>.<span class="pp">*</span><span class="va">})</span></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a>     <span class="ex">busco</span> <span class="dt">\</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a>        <span class="at">--in</span> <span class="va">${pep}</span> <span class="dt">\</span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a>        <span class="at">--cpu</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a>        <span class="at">--out</span> <span class="va">${base}</span>_busco <span class="dt">\</span></span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a>        <span class="at">--mode</span> prot <span class="dt">\</span></span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a>        <span class="at">--lineage_dataset</span> brassicales_odb10 <span class="dt">\</span></span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a>        <span class="at">--force</span></span>
<span id="cb3-17"><a href="#cb3-17" tabindex="-1"></a><span class="cf">done</span></span></code></pre>
</div>
<p>Once the BUSCO assesment is complete, we can view the results using
the following command:</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="ex">ml</span> busco</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/08_assessment</span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> busco_results</span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a><span class="cf">for</span> f <span class="kw">in</span> <span class="pp">*</span>/<span class="pp">*</span>_busco/short_summary<span class="pp">*</span>busco.txt<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a>   <span class="fu">cp</span> <span class="va">$f</span> busco_results/<span class="kw">;</span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a><span class="ex">generate_plot.py</span> –wd busco_results</span></code></pre>
</div>
</section><section><h2 class="section-heading" id="omark-assesment">Omark assesment<a class="anchor" aria-label="anchor" href="#omark-assesment"></a>
</h2>
<hr class="half-width">
<p>OMArk is a software for proteome (protein-coding gene repertoire)
quality assessment. It provides measures of proteome completeness,
characterizes the consistency of all protein coding genes with regard to
their homologs, and identifies the presence of contamination from other
species.</p>
<p>The proteomes should be filtered to take only primary isoforms before
running OMark.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="ex">ml</span> seqkit</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/08_assessment</span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${workdir}</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="cf">for</span> pep <span class="kw">in</span> braker<span class="pp">*</span>_pep.fa<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>    <span class="va">base</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">${pep</span><span class="op">%</span>.<span class="pp">*</span><span class="va">})</span></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a>    <span class="fu">grep</span> <span class="st">"\.t1$"</span> <span class="va">$pep</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/&gt;//g'</span> <span class="op">&gt;</span> <span class="va">${base}</span>.primary.ids</span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a>    <span class="ex">seqkit</span> grep <span class="at">-f</span> <span class="va">${base}</span>.primary.ids <span class="va">$pep</span> <span class="op">&gt;</span> <span class="va">${base}</span>.primary.pep.fa</span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a>    <span class="fu">rm</span> <span class="va">${base}</span>.primary.ids</span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a><span class="fu">cp</span> helixer_pep.fa helixer.primary.pep.fa</span></code></pre>
</div>
<p>Running OMark:</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a><span class="ex">ml</span> omark</span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/08_assessment</span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a><span class="va">database</span><span class="op">=</span><span class="st">"/depot/itap/datasets/omark/LUCA.h5"</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${workdir}</span></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a><span class="cf">for</span> pep <span class="kw">in</span> <span class="pp">*</span>.primary.pep.fa<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a>    <span class="va">base</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">${pep</span><span class="op">%</span>.<span class="pp">*</span><span class="va">})</span></span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a>    <span class="ex">omamer</span> search <span class="dt">\</span></span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a>       <span class="at">--db</span> <span class="va">${database}</span> <span class="dt">\</span></span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a>       <span class="at">--query</span> <span class="va">${pep}</span> <span class="dt">\</span></span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a>       <span class="at">--out</span> <span class="va">${pep</span><span class="op">%</span>.<span class="pp">*</span><span class="va">}</span>.omamer <span class="dt">\</span></span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a>       <span class="at">--nthreads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span>
<span id="cb6-16"><a href="#cb6-16" tabindex="-1"></a>    <span class="ex">omark</span> <span class="dt">\</span></span>
<span id="cb6-17"><a href="#cb6-17" tabindex="-1"></a>       <span class="at">--file</span> <span class="va">${pep</span><span class="op">%</span>.<span class="pp">*</span><span class="va">}</span>.omamer <span class="dt">\</span></span>
<span id="cb6-18"><a href="#cb6-18" tabindex="-1"></a>       <span class="at">--database</span> <span class="va">${database}</span> <span class="dt">\</span></span>
<span id="cb6-19"><a href="#cb6-19" tabindex="-1"></a>       <span class="at">--outputFolder</span> <span class="va">${pep</span><span class="op">%</span>.<span class="pp">*</span><span class="va">}</span> <span class="dt">\</span></span>
<span id="cb6-20"><a href="#cb6-20" tabindex="-1"></a>       <span class="at">--og_fasta</span> <span class="va">${pep}</span></span>
<span id="cb6-21"><a href="#cb6-21" tabindex="-1"></a><span class="cf">done</span></span></code></pre>
</div>
<div id="accordionSpoiler1" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler1" aria-expanded="false" aria-controls="collapseSpoiler1">
  <h3 class="accordion-header" id="headingSpoiler1">
  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>
  OMark Results Summary
  </h3>
</button>
<div id="collapseSpoiler1" class="accordion-collapse collapse" aria-labelledby="headingSpoiler1" data-bs-parent="#accordionSpoiler1">
<div class="accordion-body">
<p><strong>Conserved HOGs in Brassicaceae</strong></p>
<table class="table">
<thead><tr class="header">
<th align="left"><strong>Category</strong></th>
<th align="right"><strong>Count</strong></th>
<th align="center"><strong>Percentage</strong></th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><strong>Total Conserved HOGs</strong></td>
<td align="right">17,996</td>
<td align="center">100%</td>
</tr>
<tr class="even">
<td align="left"><strong>Single-Copy HOGs (S)</strong></td>
<td align="right">16,237</td>
<td align="center">90.23%</td>
</tr>
<tr class="odd">
<td align="left"><strong>Duplicated HOGs (D)</strong></td>
<td align="right">1,315</td>
<td align="center">7.31%</td>
</tr>
<tr class="even">
<td align="left"><em>Unexpected Duplications (U)</em></td>
<td align="right">351</td>
<td align="center">1.95%</td>
</tr>
<tr class="odd">
<td align="left"><em>Expected Duplications (E)</em></td>
<td align="right">964</td>
<td align="center">5.36%</td>
</tr>
<tr class="even">
<td align="left"><strong>Missing HOGs (M)</strong></td>
<td align="right">444</td>
<td align="center">2.47%</td>
</tr>
</tbody>
</table>
<p><strong>Proteome Composition</strong></p>
<table class="table">
<thead><tr class="header">
<th align="left"><strong>Category</strong></th>
<th align="right"><strong>Count</strong></th>
<th align="right"><strong>Percentage</strong></th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><strong>Total Proteins</strong></td>
<td align="right">26,994</td>
<td align="right">100%</td>
</tr>
<tr class="even">
<td align="left"><strong>Consistent (A)</strong></td>
<td align="right">25,480</td>
<td align="right">94.39%</td>
</tr>
<tr class="odd">
<td align="left"><em>Partial Hits (P)</em></td>
<td align="right">1,306</td>
<td align="right">4.84%</td>
</tr>
<tr class="even">
<td align="left"><em>Fragmented (F)</em></td>
<td align="right">630</td>
<td align="right">2.33%</td>
</tr>
<tr class="odd">
<td align="left"><strong>Inconsistent (I)</strong></td>
<td align="right">167</td>
<td align="right">0.62%</td>
</tr>
<tr class="even">
<td align="left"><em>Partial Hits (P)</em></td>
<td align="right">106</td>
<td align="right">0.39%</td>
</tr>
<tr class="odd">
<td align="left"><em>Fragmented (F)</em></td>
<td align="right">29</td>
<td align="right">0.11%</td>
</tr>
<tr class="even">
<td align="left"><strong>Likely Contamination (C)</strong></td>
<td align="right">0</td>
<td align="right">0.00%</td>
</tr>
<tr class="odd">
<td align="left"><em>Partial Hits (P)</em></td>
<td align="right">0</td>
<td align="right">0.00%</td>
</tr>
<tr class="even">
<td align="left"><em>Fragmented (F)</em></td>
<td align="right">0</td>
<td align="right">0.00%</td>
</tr>
<tr class="odd">
<td align="left"><strong>Unknown (U)</strong></td>
<td align="right">1,347</td>
<td align="right">4.99%</td>
</tr>
</tbody>
</table>
<p><strong>HOG Placement - Detected Species</strong></p>
<table class="table">
<colgroup>
<col width="53%">
<col width="15%">
<col width="15%">
<col width="15%">
</colgroup>
<thead><tr class="header">
<th><strong>Species</strong></th>
<th align="right"><strong>NCBI TaxID</strong></th>
<th align="right"><strong>Associated Proteins</strong></th>
<th align="right"><strong>% of Total Proteome</strong></th>
</tr></thead>
<tbody><tr class="odd">
<td><em>Arabidopsis thaliana</em></td>
<td align="right">3702</td>
<td align="right">25,647</td>
<td align="right">95.01%</td>
</tr></tbody>
</table>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="feature-assignment">Feature assignment<a class="anchor" aria-label="anchor" href="#feature-assignment"></a>
</h2>
<hr class="half-width">
<p>To measure the number of raw reads assigned to the features predicted
by the annotation, we will use <code>featureCounts</code> from the
<code>subread</code> package.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="ex">ml</span> subread</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/08_assessment</span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${workdir}</span></span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> featureCounts</span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a><span class="cf">for</span> gff3 <span class="kw">in</span> <span class="pp">*</span>.gff3<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a>    <span class="va">base</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">${gff3</span><span class="op">%</span>.<span class="pp">*</span><span class="va">})</span></span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a>    <span class="ex">featureCounts</span> <span class="dt">\</span></span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a>      <span class="at">-T</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a>      <span class="at">-a</span> <span class="va">${gff3}</span> <span class="dt">\</span></span>
<span id="cb7-14"><a href="#cb7-14" tabindex="-1"></a>      <span class="at">-t</span> exon <span class="dt">\</span></span>
<span id="cb7-15"><a href="#cb7-15" tabindex="-1"></a>      <span class="at">-g</span> ID <span class="dt">\</span></span>
<span id="cb7-16"><a href="#cb7-16" tabindex="-1"></a>      <span class="at">-p</span> <span class="dt">\</span></span>
<span id="cb7-17"><a href="#cb7-17" tabindex="-1"></a>      <span class="at">-B</span> <span class="dt">\</span></span>
<span id="cb7-18"><a href="#cb7-18" tabindex="-1"></a>      <span class="at">-o</span> <span class="va">${base}</span>_merged_counts.txt <span class="dt">\</span></span>
<span id="cb7-19"><a href="#cb7-19" tabindex="-1"></a>      <span class="at">--tmpDir</span> /tmp <span class="va">${RCAC_SCRATCH}</span>/annotation_workshop/00_datasets/bamfiles/<span class="pp">*</span>.bam</span>
<span id="cb7-20"><a href="#cb7-20" tabindex="-1"></a><span class="cf">done</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="reference-annotation-comparison">Reference annotation comparison<a class="anchor" aria-label="anchor" href="#reference-annotation-comparison"></a>
</h2>
<hr class="half-width">
<p>We will use <code>mikado compare</code> the reference annotation with
the predicted annotation.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a><span class="ex">ml</span> mikado</span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/08_assessment</span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${workdir}</span></span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a><span class="cf">for</span> gff3 <span class="kw">in</span> <span class="pp">*</span>.gff3<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a>    <span class="va">base</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">${gff3</span><span class="op">%</span>.<span class="pp">*</span><span class="va">})</span></span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>    <span class="ex">mikado</span> compare <span class="dt">\</span></span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a>      <span class="at">--protein-coding</span> <span class="dt">\</span></span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a>      <span class="at">-r</span> <span class="va">${RCAC_SCRATCH}</span>/annotation_workshop/00_datasets/genome/athaliana_TAIR10.gff3 <span class="dt">\</span></span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a>      <span class="at">-p</span> <span class="va">${gff3}</span> <span class="dt">\</span></span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a>      <span class="at">-o</span> ref-TAIR10_vs_prediction_<span class="va">${base}</span>_compared <span class="dt">\</span></span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a>      <span class="at">--log</span> <span class="va">${base}</span>_compare.log</span>
<span id="cb8-16"><a href="#cb8-16" tabindex="-1"></a><span class="cf">done</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="summary">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h2>
<hr class="half-width">
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>
<code>busco</code> and <code>omark</code> assess how well conserved
genes are represented in the predicted gene set<br>
</li>
<li>
<code>gff3</code> metrics provide structural insights and highlight
discrepancies compared to known annotations<br>
</li>
<li>
<code>featureCounts</code> assignment quantifies the number of
RNA-seq reads aligning to predicted features<br>
</li>
<li>Reference annotation comparison evaluates how closely the predicted
genes match an established reference<br>
</li>
<li>Multiple assessment methods ensure a comprehensive evaluation of
annotation quality</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section>
</div>
    </main>
</div>
<!-- END  : inst/pkgdown/templates/content-extra.html -->

      </div>
<!--/div.row-->
      		<footer class="row footer mx-md-3"><hr>
<div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://rcac-bioinformatics.github.io/genome-annotation/edit/main/README.md" class="external-link">Edit on GitHub</a>

	
        | <a href="https://rcac-bioinformatics.github.io/genome-annotation/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://rcac-bioinformatics.github.io/genome-annotation/" class="external-link">Source</a></p>
				<p><a href="https://rcac-bioinformatics.github.io/genome-annotation/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:rcac-help@purdue.edu">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.11" class="external-link">sandpaper (0.16.11)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.8" class="external-link">pegboard (0.7.8)</a>, and <a href="https://github.com/aseetharam/varnish/tree/db4f5c6f3c4799dfe8247fd057642c8099d5244b" class="external-link">varnish (1.0.5)</a></p>
			</div>
		</footer>
</div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://genome-annotation.github.io/NA/instructor/aio.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "gene-annotation, braker, helixer, easel, entap, omark, busco",
  "name": "All in One View",
  "creativeWorkStatus": "active",
  "url": "https://genome-annotation.github.io/NA/instructor/aio.html",
  "identifier": "https://genome-annotation.github.io/NA/instructor/aio.html",
  "dateCreated": "2025-02-05",
  "dateModified": "2025-02-22",
  "datePublished": "2025-02-22"
}

  </script><script>
		feather.replace();
	</script>
</body>
</html><!-- END:   inst/pkgdown/templates/layout.html-->

